<!DOCTYPE html><html lang="zh-CN" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width,initial-scale=1,viewport-fit=cover"><title>二代测序数据基础分析(一) | JinhengHao的博客</title><meta name="author" content="Jinheng Hao"><meta name="copyright" content="Jinheng Hao"><meta name="format-detection" content="telephone=no"><meta name="theme-color" content="#ffffff"><meta name="description" content="技术介绍 第二代测序技术，1990s - 2010s  第二代测序技术从原理上分为三种方法：  Roche 454测序法 IIIumina Solexa&#x2F;Hiseq测序法 ABI Solid测序法  三种二代测序技术的对比：  总结 二代测序技术总体而言都有着同一种特性：  需荧光或化学发光物质 需聚合酶或连接酶 需购买昂贵的试剂耗材和光学系统 需强大的图形分析计算能力  WES分析流程"><meta property="og:type" content="article"><meta property="og:title" content="二代测序数据基础分析(一)"><meta property="og:url" content="https://www.jinhenghaoblog.top/posts/20250926164634.html"><meta property="og:site_name" content="JinhengHao的博客"><meta property="og:description" content="技术介绍 第二代测序技术，1990s - 2010s  第二代测序技术从原理上分为三种方法：  Roche 454测序法 IIIumina Solexa&#x2F;Hiseq测序法 ABI Solid测序法  三种二代测序技术的对比：  总结 二代测序技术总体而言都有着同一种特性：  需荧光或化学发光物质 需聚合酶或连接酶 需购买昂贵的试剂耗材和光学系统 需强大的图形分析计算能力  WES分析流程"><meta property="og:locale" content="zh_CN"><meta property="og:image" content="https://www.jinhenghaoblog.top/img/butterfly-icon.jpg"><meta property="article:published_time" content="2025-09-26T08:46:34.000Z"><meta property="article:modified_time" content="2025-09-28T13:17:07.155Z"><meta property="article:author" content="Jinheng Hao"><meta property="article:tag" content="基因测序"><meta name="twitter:card" content="summary"><meta name="twitter:image" content="https://www.jinhenghaoblog.top/img/butterfly-icon.jpg"><script type="application/ld+json">{
  "@context": "https://schema.org",
  "@type": "BlogPosting",
  "headline": "二代测序数据基础分析(一)",
  "url": "https://www.jinhenghaoblog.top/posts/20250926164634.html",
  "image": "https://www.jinhenghaoblog.top/img/butterfly-icon.jpg",
  "datePublished": "2025-09-26T08:46:34.000Z",
  "dateModified": "2025-09-28T13:17:07.155Z",
  "author": [
    {
      "@type": "Person",
      "name": "Jinheng Hao",
      "url": "https://www.jinhenghaoblog.top"
    }
  ]
}</script><link rel="shortcut icon" href="/images/favicon.ico"><link rel="canonical" href="https://www.jinhenghaoblog.top/posts/20250926164634.html"><link rel="preconnect" href="//cdn.jsdelivr.net"><link rel="preconnect" href="//busuanzi.ibruce.info"><meta name="google-site-verification" content="YCuKDW04QOezpE8jWTLZ_yiyayYEZjHGT3P_qo9C13I"><meta name="baidu-site-verification" content="codeva-Wv2xaBiiDf"><link rel="stylesheet" href="/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free/css/all.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/node-snackbar/dist/snackbar.min.css" media="print" onload='this.media="all"'><script>(()=>{var e={set:(e,t,a)=>{a&&(a=Date.now()+864e5*a,localStorage.setItem(e,JSON.stringify({value:t,expiry:a})))},get:e=>{var t=localStorage.getItem(e);if(t){var{value:t,expiry:a}=JSON.parse(t);if(!(Date.now()>a))return t;localStorage.removeItem(e)}}},t=(window.btf={saveToLocal:e,getScript:(o,n={})=>new Promise((e,t)=>{let a=document.createElement("script");a.src=o,a.async=!0,Object.entries(n).forEach(([e,t])=>a.setAttribute(e,t)),a.onload=a.onreadystatechange=()=>{a.readyState&&!/loaded|complete/.test(a.readyState)||e()},a.onerror=t,document.head.appendChild(a)}),getCSS:(o,n)=>new Promise((e,t)=>{let a=document.createElement("link");a.rel="stylesheet",a.href=o,n&&(a.id=n),a.onload=a.onreadystatechange=()=>{a.readyState&&!/loaded|complete/.test(a.readyState)||e()},a.onerror=t,document.head.appendChild(a)}),addGlobalFn:(e,t,a=!1,o=window)=>{var n;e.startsWith("pjax")||((n=o.globalFn||{})[e]=n[e]||{},n[e][a||Object.keys(n[e]).length]=t,o.globalFn=n)}},()=>{document.documentElement.setAttribute("data-theme","dark"),null!==document.querySelector('meta[name="theme-color"]')&&document.querySelector('meta[name="theme-color"]').setAttribute("content","#0d0d0d")}),a=()=>{document.documentElement.setAttribute("data-theme","light"),null!==document.querySelector('meta[name="theme-color"]')&&document.querySelector('meta[name="theme-color"]').setAttribute("content","#ffffff")},o=(btf.activateDarkMode=t,btf.activateLightMode=a,e.get("theme")),t=("dark"===o?t():"light"===o&&a(),e.get("aside-status"));void 0!==t&&document.documentElement.classList.toggle("hide-aside","hide"===t);/iPad|iPhone|iPod|Macintosh/.test(navigator.userAgent)&&document.documentElement.classList.add("apple")})()</script><script>let GLOBAL_CONFIG={root:"/",algolia:void 0,localSearch:void 0,translate:void 0,highlight:{plugin:"highlight.js",highlightCopy:!0,highlightLang:!0,highlightHeightLimit:!1,highlightFullpage:!1,highlightMacStyle:!1},copy:{success:"复制成功",error:"复制失败",noSupport:"浏览器不支持"},relativeDate:{homepage:!1,post:!1},runtime:"",dateSuffix:{just:"刚刚",min:"分钟前",hour:"小时前",day:"天前",month:"个月前"},copyright:void 0,lightbox:"null",Snackbar:{chs_to_cht:"已切换为繁体中文",cht_to_chs:"已切换为简体中文",day_to_night:"已切换为深色模式",night_to_day:"已切换为浅色模式",bgLight:"#49b1f5",bgDark:"#1f1f1f",position:"bottom-left"},infinitegrid:{js:"https://cdn.jsdelivr.net/npm/@egjs/infinitegrid/dist/infinitegrid.min.js",buttonText:"加载更多"},isPhotoFigcaption:!1,islazyloadPlugin:!1,isAnchor:!1,percent:{toc:!0,rightside:!1},autoDarkmode:!1}</script><script id="config-diff">var GLOBAL_CONFIG_SITE={title:"二代测序数据基础分析(一)",isHighlightShrink:!1,isToc:!0,pageType:"post"}</script><meta name="generator" content="Hexo 7.3.0"><link rel="alternate" href="/atom.xml" title="JinhengHao的博客" type="application/atom+xml"></head><body><div id="sidebar"><div id="menu-mask"></div><div id="sidebar-menus"><div class="avatar-img text-center"><img src="/img/butterfly-icon.jpg" onerror='this.onerror=null,this.src="/img/friend_404.gif"' alt="avatar"></div><div class="site-data text-center"><a href="/archives/"><div class="headline">文章</div><div class="length-num">22</div></a><a href="/tags/"><div class="headline">标签</div><div class="length-num">13</div></a><a href="/categories/"><div class="headline">分类</div><div class="length-num">7</div></a></div><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fas fa-home"></i><span> 主页</span></a></div><div class="menus_item"><a class="site-page" href="/search/"><i class="fa-fw fas fa-folder-open"></i><span> 搜索</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fas fa-folder-open"></i><span> 分类</span></a></div><div class="menus_item"><a class="site-page" href="/about/"><i class="fa-fw fas fa-folder-open"></i><span> 关于</span></a></div></div></div></div><div class="post" id="body-wrap"><header class="post-bg" id="page-header" style="background-image:url(/img/background.png)"><nav id="nav"><span id="blog-info"><a class="nav-site-title" href="/"><img class="site-icon" src="/img/butterfly-icon.jpg" alt="Logo"><span class="site-name">JinhengHao的博客</span></a><a class="nav-page-title" href="/"><span class="site-name">二代测序数据基础分析(一)</span><span class="site-name"><i class="fa-solid fa-circle-arrow-left"></i><span> 返回首页</span></span></a></span><div id="menus"><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fas fa-home"></i><span> 主页</span></a></div><div class="menus_item"><a class="site-page" href="/search/"><i class="fa-fw fas fa-folder-open"></i><span> 搜索</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fas fa-folder-open"></i><span> 分类</span></a></div><div class="menus_item"><a class="site-page" href="/about/"><i class="fa-fw fas fa-folder-open"></i><span> 关于</span></a></div></div><div id="toggle-menu"><span class="site-page"><i class="fas fa-bars fa-fw"></i></span></div></div></nav><div id="post-info"><h1 class="post-title">二代测序数据基础分析(一)</h1><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">发表于</span><time class="post-meta-date-created" datetime="2025-09-26T08:46:34.000Z" title="发表于 2025-09-26 16:46:34">2025-09-26</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">更新于</span><time class="post-meta-date-updated" datetime="2025-09-28T13:17:07.155Z" title="更新于 2025-09-28 21:17:07">2025-09-28</time></span><span class="post-meta-categories"><span class="post-meta-separator">|</span><i class="fas fa-inbox fa-fw post-meta-icon"></i><a class="post-meta-categories" href="/categories/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF/">生物信息</a><i class="fas fa-angle-right post-meta-separator"></i><i class="fas fa-inbox fa-fw post-meta-icon"></i><a class="post-meta-categories" href="/categories/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF/%E5%AD%A6%E4%B9%A0/">学习</a></span></div><div class="meta-secondline"><span class="post-meta-separator">|</span><span data-flag-title=""><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">浏览量:</span><span id="twikoo_visitors"><i class="fa-solid fa-spinner fa-spin"></i></span></span><span class="post-meta-separator">|</span><span class="post-meta-commentcount"><i class="far fa-comments fa-fw post-meta-icon"></i><span class="post-meta-label">评论数:</span><a href="/posts/20250926164634.html#post-comment"><span id="twikoo-count"><i class="fa-solid fa-spinner fa-spin"></i></span></a></span></div></div></div></header><main class="layout" id="content-inner"><div id="post"><article class="container post-content" id="article-container"><h2 id="技术介绍"><a href="#技术介绍" class="headerlink" title="技术介绍"></a>技术介绍</h2><ul><li>第二代测序技术，1990s - 2010s</li></ul><p>第二代测序技术从原理上分为三种方法：</p><ul><li>Roche 454测序法</li><li>IIIumina Solexa&#x2F;Hiseq测序法</li><li>ABI Solid测序法</li></ul><p>三种二代测序技术的对比：</p><p><img src="/images/image_1758975659796.png" alt="image.png"></p><p><strong>总结</strong></p><p>二代测序技术总体而言都有着同一种特性：</p><ul><li>需荧光或化学发光物质</li><li>需聚合酶或连接酶</li><li>需购买昂贵的试剂耗材和光学系统</li><li>需强大的图形分析计算能力</li></ul><h2 id="WES分析流程"><a href="#WES分析流程" class="headerlink" title="WES分析流程"></a>WES分析流程</h2><p>WES分析流程分为5个步骤：</p><ol><li>WES分析流程</li><li>数据质量控制</li><li>序列比对分析</li><li>变异检测</li><li>变异注释</li></ol><h3 id="一-WES分析流程"><a href="#一-WES分析流程" class="headerlink" title="一. WES分析流程"></a>一. WES分析流程</h3><p>原始测序数据——数据质量控制——序列比对——变异检测——变异注释——完成</p><h3 id="二-数据质量控制"><a href="#二-数据质量控制" class="headerlink" title="二. 数据质量控制"></a>二. 数据质量控制</h3><p>数据质控标准：</p><ol><li>去掉reads中的接头</li><li>去除低质量（BP&lt;20）碱基的reads</li><li>去除序列头尾的N碱基</li><li>去除头尾N碱基后若剩余reads长度小于40bp（双端），则丢弃该对序列</li></ol><p><strong>原始数据下载</strong>——sratoolkit软件</p><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta prompt_"># </span><span class="language-bash">安装</span></span><br><span class="line">conda install sra-tools</span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">找到sra数据，下载srr list</span></span><br><span class="line">prefetch SRR1139956</span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">sra转为fastq</span></span><br><span class="line">nohup fastq-dump -gzip --split-3 -O ./ SRR1139956.sra &amp;</span><br></pre></td></tr></table></figure><p>基本参数：</p><p>fastq-dump [options] [path]</p><ol><li>–gzip: 使用gzip压缩输出文件</li><li>–split-3: 设置文件拆分，如果是双端的序列，则拆分为*_1.fastq和*_2.fastq</li><li>-O: 设置输出文件路径</li></ol><p>数据质量控制常用的软件是FASTQC，在这里不再赘述，详情请看<a href="https://" rel="external nofollow noreferrer">https://www.jinhenghaoblog.top/posts/20250828165237.html</a></p><h3 id="序列比对分析"><a href="#序列比对分析" class="headerlink" title="序列比对分析"></a>序列比对分析</h3><p><strong>进行序列比对分析所用到的软件为BWA</strong>：</p><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta prompt_"># </span><span class="language-bash">下载参考数据</span></span><br><span class="line">wget ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg38/Homo_sapiens_assembly38.fasta.gz</span><br><span class="line">gunzip Homo_sapiens_assembly38.fasta.gz</span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">建立索引</span></span><br><span class="line">bwa index Homo_sapiens_assembly38.fasta</span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">bwa进行比对</span></span><br><span class="line">nohup bwa mem -t 5 db/Homo_sapiens_assembly38.fasta test.fastq 1&gt;test.sam 2&gt;test.log</span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">这里的test.fastq是经过过滤得到的fastq文件</span></span><br></pre></td></tr></table></figure><p>基本参数：</p><p>bwa mem [options] [idxbase] [in1.fq] [in2.fq]</p><ol><li>idxbase: 指定比对参考基因组序列（建好索引）</li><li>in1.fq、in2.fq：输入fastq文件</li><li>-t : 设置线程，这里设置的是5，具体看计算机配置</li></ol><p><strong>SAM文件格式转换及排序</strong></p><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta prompt_"># </span><span class="language-bash">sam转bam</span></span><br><span class="line">samtool view -bS -h test.sam &gt; test.bam</span><br><span class="line"><span class="meta prompt_"></span></span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">排序bam</span></span><br><span class="line">samtools sort -@ 5 test.bam -o test.sort.bam</span><br></pre></td></tr></table></figure><p>基本参数：</p><p>samtools sort [options] [in.bam] [out.prefix]</p><ol><li>-n: 根据reads名字排序</li><li>-o: 输出文件</li><li>-@: 设置线程，这里设置的是5，具体看计算机配置</li></ol><p><strong>标记&#x2F;删除PCR重复的reads——GATK</strong></p><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta prompt_"># </span><span class="language-bash">建立输出文件夹</span></span><br><span class="line">mkdir gatk_markdup</span><br><span class="line">cd gatk_markdup</span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">标记PCR重复的reads</span></span><br><span class="line">gatk --java-options &quot;-Xmx20G -Djava.io.tmpdir=./&quot; MarkDuplicates -I test.sort.bam -M test.metrics 1&gt;test_log.mark 2 &gt;&amp;1</span><br></pre></td></tr></table></figure><p>基本参数：</p><p>gatk MarkDuplicates -I [INPUT] -O [OUTPUT] -M [METRICS]</p><ol><li>MarkDuplicates: 鉴定重复的reads，是GATK中的一个工具</li><li>-I ： 输入文件</li><li>-O ： 输出文件</li><li>-M： 复制度量写入的文件</li></ol><h3 id="变异检测"><a href="#变异检测" class="headerlink" title="变异检测"></a>变异检测</h3><p><strong>变异检测——标记flagstat</strong></p><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">gatk --java-options &quot;-Xmx20G -Djava.io.tmpdir=./&quot; FixMateInformation -I test_marked.bam -O test_marked_fixed.bam -SO coordinate 1&gt;test_log.fix 2&gt;&amp;1</span><br><span class="line">samtools index test_marked_fixed.bam</span><br></pre></td></tr></table></figure><p>基本参数：</p><p>gatk FixMateInformation -I [INPUT] -O [OUTPUT] -M [METRICS]</p><ol><li>FixMateInformation : 是GATK中的一个工具</li><li>-I：输入文件</li><li>-O：输出文件</li><li>-SO： 输出文件的排序方式，这里选择的是coordinate</li></ol><p><strong>变异检测——GATK找变异</strong></p><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta prompt_"># </span><span class="language-bash">找变异</span></span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">下载参考</span></span><br><span class="line">nohup wget ftp://gsapubftp-anonymous@ftp,broadinstitute.org/bundle/hg38/dbsnp 146.hg38.vcf.gz &amp;</span><br><span class="line">nohup wget ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg38/Mills and 1000G gold standard.indels.hg38.vcf.gz &amp;</span><br><span class="line"></span><br><span class="line">indel=&quot;Mills_and_1000G_gold_standard.indels.hg38.vcf.gz&quot;</span><br><span class="line">snp=&quot;dbsnp_146.hg38.vcf.gz&quot;</span><br><span class="line"><span class="meta prompt_"></span></span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">运行命令</span></span><br><span class="line">gatk --java-options &quot;-Xmx20G -Djava.io.tmpdir=./&quot; BaseRecalibrator -R /analysis/Database/hg38/hg38.fa -I test_marked_fixed.bam --known-sites $snp --known-sites $indel -O test_recal.table</span><br></pre></td></tr></table></figure><p>基本参数：</p><p>gatk BaseRecalibrator -I [INPUT] -O [OUTPUT] -M [METRICS]</p><ol><li>–known-sites: 已知的多态位点数据库（可以指定多个）</li><li>-R： 参考序列文件</li><li>-I：输入文件</li><li>-O：输出文件</li></ol><p><strong>变异检测——矫正bam并生成VCF</strong></p><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta prompt_"># </span><span class="language-bash">————————矫正bam</span></span><br><span class="line">gatk --java-options &quot;-XmX20G -Djava.io.tmpdir=./&quot; ApplyBQSR -R /analysis/Database/hg38/hg38.fa -I test_marked_fixed.bam -bqsr test_recal_table -O test_bqsr.bam 1&gt;test_log.ApplyBQSR 2&gt;&amp;1</span><br><span class="line"><span class="meta prompt_"></span></span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">——————calling variation得到vcf</span></span><br><span class="line">gatk --java-options &quot;-Xmx20G -Djava.io.tmpdir=./&quot; HaplotypeCaller -R /analysis/Database/hg38/hg38.fa -I test_bqsr.bam --dbsnp $snp -O test_raw.vcf 1&gt;test_log.HC 2&gt;&amp;1</span><br><span class="line">samtools index test.bqsr.bam</span><br></pre></td></tr></table></figure><p>基本参数：</p><p>gatk HaplotypeCaller -i [INPUT] -O [OUTPUT] -M [METRICS]</p><ol><li>-R：参考序列文件</li><li>-I：输入文件</li><li>–dbsnp：dbSNP文件</li><li>-O：输出文件</li></ol><h3 id="变异注释"><a href="#变异注释" class="headerlink" title="变异注释"></a>变异注释</h3><p>变异注释使用的软件为Annovar软件</p><p>Annovar软件可实现三种不同的注释方式：</p><ul><li>确定SNP或CNV是否导致蛋白质编码变化和确定受影响的氨基酸</li><li>识别特定基因组区域的变异</li><li>鉴定特定数据库中记录的变异</li></ul><p><strong>Annovar——下载数据库</strong></p><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">perl annotate_variation.pl -buildver hg19 -downdb -webfrom annovar refGene humandb/</span><br></pre></td></tr></table></figure><p>基本参数：</p><p>Perl annotate_variation.pl [options]</p><ol><li>-buildver: 基因组对应版本，这里是hg19</li><li>-webfrom annovar：指定从annovar库里下载，若annovar库中没有，则会从UCSC中下载</li><li>refGene：数据库名称</li><li>humandb&#x2F;：数据库存储路径</li></ol><p><strong>Annovar——变异注释</strong></p><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta prompt_"># </span><span class="language-bash">输入文件格式转换</span></span><br><span class="line">perl conver2annovar.pl -format vcf4 -allsample test.indel.vcf.gz -outfile anno</span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">注释</span></span><br><span class="line">perl annotate_variation.pl -generanno -buildver hg38 test.avinput humandb/</span><br></pre></td></tr></table></figure><p>基本参数：</p><p>Perl table_annovar.pl [options]</p><ol><li>-remove: 删除所有的临时文件</li><li>-geneanno：指定注释方法。这里表示gene-based，-regionanno表示region-based，–filer表示filter-based；</li><li>-buildver: 指定参考基因组</li></ol></article><div class="tag_share"><div class="post-meta__tag-list"><a class="post-meta__tags" href="/tags/%E5%9F%BA%E5%9B%A0%E6%B5%8B%E5%BA%8F/">基因测序</a></div><div class="post-share"><div class="social-share" data-image="/img/butterfly-icon.jpg" data-sites="wechat,weibo,qq"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/butterfly-extsrc/sharejs/dist/css/share.min.css" media="print" onload='this.media="all"'><script src="https://cdn.jsdelivr.net/npm/butterfly-extsrc/sharejs/dist/js/social-share.min.js" defer></script></div></div><nav class="pagination-post" id="pagination"><a class="pagination-related full-width" href="/posts/20250924211902.html" title="python导入模块时遇到ModuleNotFoundError: No module named ’xxx‘"><div class="cover" style="background:var(--default-bg-color)"></div><div class="info text-right"><div class="info-1"><div class="info-item-1">下一篇</div><div class="info-item-2">python导入模块时遇到ModuleNotFoundError: No module named ’xxx‘</div></div><div class="info-2"><div class="info-item-1">前言记录以前学习python的时候遇到模块导入问题的解决方法 问题出现过程在python导入模块时候遇到 ModuleNotFoundError: No module named “pandas” 出现错误的代码块如下： 12import pandas as pdimport numpy as np 出现： 123Traceback (most recent call last): File &quot;&lt;stdin&gt;&quot;, line 1, in &lt;module&gt;ModuleNotFoundError: No module named &#x27;pandas&#x27; 解决方法在terminal下运行指令： 12345678# 使用condaconda install pandas# 使用pippip install pandas# 在该环境下使用pip安装python -m pip install pandas# 基于清华源使用pip安装（推荐）pip install -i https://mirrors.tuna.tsinghua.e...</div></div></div></a></nav><hr class="custom-hr"><div id="post-comment"><div class="comment-head"><div class="comment-headline"><i class="fas fa-comments fa-fw"></i><span> 评论</span></div></div><div class="comment-wrap"><div><div id="twikoo-wrap"></div></div></div></div></div><div class="aside-content" id="aside-content"><div class="card-widget card-info text-center"><div class="avatar-img"><img src="/img/butterfly-icon.jpg" onerror='this.onerror=null,this.src="/img/friend_404.gif"' alt="avatar"></div><div class="author-info-name">Jinheng Hao</div><div class="author-info-description">生信/实验/技术交流</div><div class="site-data"><a href="/archives/"><div class="headline">文章</div><div class="length-num">22</div></a><a href="/tags/"><div class="headline">标签</div><div class="length-num">13</div></a><a href="/categories/"><div class="headline">分类</div><div class="length-num">7</div></a></div><a id="card-info-btn" target="_blank" rel="noopener external nofollow noreferrer" href="https://github.com/QDcvd"><i class="fab fa-github"></i><span>Follow Me</span></a></div><div class="card-widget card-announcement"><div class="item-headline"><i class="fas fa-bullhorn fa-shake"></i><span>公告</span></div><div class="announcement_content">如果你对内容有什么疑问，请积极评论</div></div><div class="sticky_layout"><div class="card-widget" id="card-toc"><div class="item-headline"><i class="fas fa-stream"></i><span>目录</span><span class="toc-percentage"></span></div><div class="toc-content"><ol class="toc"><li class="toc-item toc-level-2"><a class="toc-link" href="#%E6%8A%80%E6%9C%AF%E4%BB%8B%E7%BB%8D"><span class="toc-number">1.</span> <span class="toc-text">技术介绍</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#WES%E5%88%86%E6%9E%90%E6%B5%81%E7%A8%8B"><span class="toc-number">2.</span> <span class="toc-text">WES分析流程</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E4%B8%80-WES%E5%88%86%E6%9E%90%E6%B5%81%E7%A8%8B"><span class="toc-number">2.1.</span> <span class="toc-text">一. WES分析流程</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E4%BA%8C-%E6%95%B0%E6%8D%AE%E8%B4%A8%E9%87%8F%E6%8E%A7%E5%88%B6"><span class="toc-number">2.2.</span> <span class="toc-text">二. 数据质量控制</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E5%BA%8F%E5%88%97%E6%AF%94%E5%AF%B9%E5%88%86%E6%9E%90"><span class="toc-number">2.3.</span> <span class="toc-text">序列比对分析</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E5%8F%98%E5%BC%82%E6%A3%80%E6%B5%8B"><span class="toc-number">2.4.</span> <span class="toc-text">变异检测</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E5%8F%98%E5%BC%82%E6%B3%A8%E9%87%8A"><span class="toc-number">2.5.</span> <span class="toc-text">变异注释</span></a></li></ol></li></ol></div></div><div class="card-widget card-recent-post"><div class="item-headline"><i class="fas fa-history"></i><span>最新文章</span></div><div class="aside-list"><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/posts/20250926164634.html" title="二代测序数据基础分析(一)">二代测序数据基础分析(一)</a><time datetime="2025-09-26T08:46:34.000Z" title="发表于 2025-09-26 16:46:34">2025-09-26</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/posts/20250924211902.html" title="python导入模块时遇到ModuleNotFoundError: No module named ’xxx‘">python导入模块时遇到ModuleNotFoundError: No module named ’xxx‘</a><time datetime="2025-09-24T13:19:02.000Z" title="发表于 2025-09-24 21:19:02">2025-09-24</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/posts/20250921164632.html" title="秀丽隐杆线虫油红O响应值测定方法">秀丽隐杆线虫油红O响应值测定方法</a><time datetime="2025-09-21T08:46:32.000Z" title="发表于 2025-09-21 16:46:32">2025-09-21</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/posts/20250916200356.html" title="基于David数据库与微生信平台进行Go-KEGG分析">基于David数据库与微生信平台进行Go-KEGG分析</a><time datetime="2025-09-16T12:03:56.000Z" title="发表于 2025-09-16 20:03:56">2025-09-16</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/posts/20250903205237.html" title="RNA-seq基本介绍及实战——差异分析">RNA-seq基本介绍及实战——差异分析</a><time datetime="2025-09-03T12:52:37.000Z" title="发表于 2025-09-03 20:52:37">2025-09-03</time></div></div></div></div></div></div></main><footer id="footer"><div class="footer-other"><div class="footer-copyright"></div></div></footer></div><div id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="阅读模式"><i class="fas fa-book-open"></i></button><button id="darkmode" type="button" title="日间和夜间模式切换"><i class="fas fa-adjust"></i></button><button id="hide-aside-btn" type="button" title="单栏和双栏切换"><i class="fas fa-arrows-alt-h"></i></button></div><div id="rightside-config-show"><button id="rightside-config" type="button" title="设置"><i class="fas fa-cog fa-spin"></i></button><button class="close" id="mobile-toc-button" type="button" title="目录"><i class="fas fa-list-ul"></i></button><a id="to_comment" href="#post-comment" title="前往评论"><i class="fas fa-comments"></i></a><button id="go-up" type="button" title="回到顶部"><span class="scroll-percent"></span><i class="fas fa-arrow-up"></i></button></div></div><div><script src="/js/utils.js"></script><script src="/js/main.js"></script><script src="https://cdn.jsdelivr.net/npm/node-snackbar/dist/snackbar.min.js"></script><div class="js-pjax"><script>(()=>{let e="shuoshuo"===GLOBAL_CONFIG_SITE.pageType,n=null,i=()=>{let t=document.getElementById("twikoo-count");t&&twikoo.getCommentsCount({envId:"https://blogcomment0516.netlify.app/.netlify/functions/twikoo",region:"",urls:[window.location.pathname],includeReply:!1}).then(o=>{t.textContent=o[0].count}).catch(o=>{console.error(o)})},l=(o=document,t=location.pathname)=>{twikoo.init({el:o.querySelector("#twikoo-wrap"),envId:"https://blogcomment0516.netlify.app/.netlify/functions/twikoo",region:"",onCommentLoaded:()=>{btf.loadLightbox(document.querySelectorAll("#twikoo .tk-content img:not(.tk-owo-emotion)"))},...n,path:!e&&n&&n.path||t}),"post"===GLOBAL_CONFIG_SITE.pageType&&i(),e&&(window.shuoshuoComment.destroyTwikoo=()=>{o.children.length&&(o.innerHTML="",o.classList.add("no-comment"))})};var o=(o,t)=>{"object"==typeof twikoo?setTimeout(()=>l(o,t),0):btf.getScript("https://cdn.jsdelivr.net/npm/twikoo/dist/twikoo.all.min.js").then(()=>l(o,t))};e?window.shuoshuoComment={loadComment:o}:o()})()</script></div><canvas class="fireworks" mobile="false"></canvas><script src="https://cdn.jsdelivr.net/npm/butterfly-extsrc/dist/fireworks.min.js"></script><script defer id="ribbon" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc/dist/canvas-ribbon.min.js" size="150" alpha="0.6" zindex="-1" mobile="false" data-click="false"></script><script async data-pjax src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script></div></body></html>