<!DOCTYPE html><html lang="zh-CN" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width,initial-scale=1,viewport-fit=cover"><title>RNA-seq基本介绍及实战——表达定量 | JinhengHao的博客</title><meta name="author" content="Jinheng Hao"><meta name="copyright" content="Jinheng Hao"><meta name="format-detection" content="telephone=no"><meta name="theme-color" content="#ffffff"><meta name="description" content="实战环节上一期对所拥有的数据做了个序列比对，现在进入表达定量的环节 表达定量在进行表达定量的处理之前，需要对原始的比对文件进行处理，这里就有以下步骤：  使用picard &#x2F; samtools 将sam格式转换为bam格式 对bam文件进行排序 去除比对得分较低的序列 如果需要可以去除重复reads  在这里将会以三种方法进行表达定量的操作，分别是STAR+RSEM进行表达定量，另外一个"><meta property="og:type" content="article"><meta property="og:title" content="RNA-seq基本介绍及实战——表达定量"><meta property="og:url" content="https://www.jinhenghaoblog.top/posts/20250831205237.html"><meta property="og:site_name" content="JinhengHao的博客"><meta property="og:description" content="实战环节上一期对所拥有的数据做了个序列比对，现在进入表达定量的环节 表达定量在进行表达定量的处理之前，需要对原始的比对文件进行处理，这里就有以下步骤：  使用picard &#x2F; samtools 将sam格式转换为bam格式 对bam文件进行排序 去除比对得分较低的序列 如果需要可以去除重复reads  在这里将会以三种方法进行表达定量的操作，分别是STAR+RSEM进行表达定量，另外一个"><meta property="og:locale" content="zh_CN"><meta property="og:image" content="https://www.jinhenghaoblog.top/img/butterfly-icon.jpg"><meta property="article:published_time" content="2025-08-31T12:52:37.000Z"><meta property="article:modified_time" content="2025-09-04T10:00:24.520Z"><meta property="article:author" content="Jinheng Hao"><meta property="article:tag" content="R"><meta property="article:tag" content="python"><meta property="article:tag" content="转录组学"><meta name="twitter:card" content="summary"><meta name="twitter:image" content="https://www.jinhenghaoblog.top/img/butterfly-icon.jpg"><script type="application/ld+json">{
  "@context": "https://schema.org",
  "@type": "BlogPosting",
  "headline": "RNA-seq基本介绍及实战——表达定量",
  "url": "https://www.jinhenghaoblog.top/posts/20250831205237.html",
  "image": "https://www.jinhenghaoblog.top/img/butterfly-icon.jpg",
  "datePublished": "2025-08-31T12:52:37.000Z",
  "dateModified": "2025-09-04T10:00:24.520Z",
  "author": [
    {
      "@type": "Person",
      "name": "Jinheng Hao",
      "url": "https://www.jinhenghaoblog.top"
    }
  ]
}</script><link rel="shortcut icon" href="/images/favicon.ico"><link rel="canonical" href="https://www.jinhenghaoblog.top/posts/20250831205237.html"><link rel="preconnect" href="//cdn.jsdelivr.net"><link rel="preconnect" href="//busuanzi.ibruce.info"><meta name="google-site-verification" content="YCuKDW04QOezpE8jWTLZ_yiyayYEZjHGT3P_qo9C13I"><meta name="baidu-site-verification" content="codeva-Wv2xaBiiDf"><link rel="stylesheet" href="/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free/css/all.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/node-snackbar/dist/snackbar.min.css" media="print" onload='this.media="all"'><script>(()=>{var e={set:(e,t,a)=>{a&&(a=Date.now()+864e5*a,localStorage.setItem(e,JSON.stringify({value:t,expiry:a})))},get:e=>{var t=localStorage.getItem(e);if(t){var{value:t,expiry:a}=JSON.parse(t);if(!(Date.now()>a))return t;localStorage.removeItem(e)}}},t=(window.btf={saveToLocal:e,getScript:(o,n={})=>new Promise((e,t)=>{let a=document.createElement("script");a.src=o,a.async=!0,Object.entries(n).forEach(([e,t])=>a.setAttribute(e,t)),a.onload=a.onreadystatechange=()=>{a.readyState&&!/loaded|complete/.test(a.readyState)||e()},a.onerror=t,document.head.appendChild(a)}),getCSS:(o,n)=>new Promise((e,t)=>{let a=document.createElement("link");a.rel="stylesheet",a.href=o,n&&(a.id=n),a.onload=a.onreadystatechange=()=>{a.readyState&&!/loaded|complete/.test(a.readyState)||e()},a.onerror=t,document.head.appendChild(a)}),addGlobalFn:(e,t,a=!1,o=window)=>{var n;e.startsWith("pjax")||((n=o.globalFn||{})[e]=n[e]||{},n[e][a||Object.keys(n[e]).length]=t,o.globalFn=n)}},()=>{document.documentElement.setAttribute("data-theme","dark"),null!==document.querySelector('meta[name="theme-color"]')&&document.querySelector('meta[name="theme-color"]').setAttribute("content","#0d0d0d")}),a=()=>{document.documentElement.setAttribute("data-theme","light"),null!==document.querySelector('meta[name="theme-color"]')&&document.querySelector('meta[name="theme-color"]').setAttribute("content","#ffffff")},o=(btf.activateDarkMode=t,btf.activateLightMode=a,e.get("theme")),t=("dark"===o?t():"light"===o&&a(),e.get("aside-status"));void 0!==t&&document.documentElement.classList.toggle("hide-aside","hide"===t);/iPad|iPhone|iPod|Macintosh/.test(navigator.userAgent)&&document.documentElement.classList.add("apple")})()</script><script>let GLOBAL_CONFIG={root:"/",algolia:void 0,localSearch:void 0,translate:void 0,highlight:{plugin:"highlight.js",highlightCopy:!0,highlightLang:!0,highlightHeightLimit:!1,highlightFullpage:!1,highlightMacStyle:!1},copy:{success:"复制成功",error:"复制失败",noSupport:"浏览器不支持"},relativeDate:{homepage:!1,post:!1},runtime:"",dateSuffix:{just:"刚刚",min:"分钟前",hour:"小时前",day:"天前",month:"个月前"},copyright:void 0,lightbox:"null",Snackbar:{chs_to_cht:"已切换为繁体中文",cht_to_chs:"已切换为简体中文",day_to_night:"已切换为深色模式",night_to_day:"已切换为浅色模式",bgLight:"#49b1f5",bgDark:"#1f1f1f",position:"bottom-left"},infinitegrid:{js:"https://cdn.jsdelivr.net/npm/@egjs/infinitegrid/dist/infinitegrid.min.js",buttonText:"加载更多"},isPhotoFigcaption:!1,islazyloadPlugin:!1,isAnchor:!1,percent:{toc:!0,rightside:!1},autoDarkmode:!1}</script><script id="config-diff">var GLOBAL_CONFIG_SITE={title:"RNA-seq基本介绍及实战——表达定量",isHighlightShrink:!1,isToc:!0,pageType:"post"}</script><meta name="generator" content="Hexo 7.3.0"><link rel="alternate" href="/atom.xml" title="JinhengHao的博客" type="application/atom+xml"></head><body><div id="sidebar"><div id="menu-mask"></div><div id="sidebar-menus"><div class="avatar-img text-center"><img src="/img/butterfly-icon.jpg" onerror='this.onerror=null,this.src="/img/friend_404.gif"' alt="avatar"></div><div class="site-data text-center"><a href="/archives/"><div class="headline">文章</div><div class="length-num">23</div></a><a href="/tags/"><div class="headline">标签</div><div class="length-num">14</div></a><a href="/categories/"><div class="headline">分类</div><div class="length-num">7</div></a></div><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fas fa-home"></i><span> 主页</span></a></div><div class="menus_item"><a class="site-page" href="/search/"><i class="fa-fw fas fa-folder-open"></i><span> 搜索</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fas fa-folder-open"></i><span> 分类</span></a></div><div class="menus_item"><a class="site-page" href="/about/"><i class="fa-fw fas fa-folder-open"></i><span> 关于</span></a></div></div></div></div><div class="post" id="body-wrap"><header class="post-bg" id="page-header" style="background-image:url(/img/background.png)"><nav id="nav"><span id="blog-info"><a class="nav-site-title" href="/"><img class="site-icon" src="/img/butterfly-icon.jpg" alt="Logo"><span class="site-name">JinhengHao的博客</span></a><a class="nav-page-title" href="/"><span class="site-name">RNA-seq基本介绍及实战——表达定量</span><span class="site-name"><i class="fa-solid fa-circle-arrow-left"></i><span> 返回首页</span></span></a></span><div id="menus"><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fas fa-home"></i><span> 主页</span></a></div><div class="menus_item"><a class="site-page" href="/search/"><i class="fa-fw fas fa-folder-open"></i><span> 搜索</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fas fa-folder-open"></i><span> 分类</span></a></div><div class="menus_item"><a class="site-page" href="/about/"><i class="fa-fw fas fa-folder-open"></i><span> 关于</span></a></div></div><div id="toggle-menu"><span class="site-page"><i class="fas fa-bars fa-fw"></i></span></div></div></nav><div id="post-info"><h1 class="post-title">RNA-seq基本介绍及实战——表达定量</h1><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">发表于</span><time class="post-meta-date-created" datetime="2025-08-31T12:52:37.000Z" title="发表于 2025-08-31 20:52:37">2025-08-31</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">更新于</span><time class="post-meta-date-updated" datetime="2025-09-04T10:00:24.520Z" title="更新于 2025-09-04 18:00:24">2025-09-04</time></span><span class="post-meta-categories"><span class="post-meta-separator">|</span><i class="fas fa-inbox fa-fw post-meta-icon"></i><a class="post-meta-categories" href="/categories/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF/">生物信息</a><i class="fas fa-angle-right post-meta-separator"></i><i class="fas fa-inbox fa-fw post-meta-icon"></i><a class="post-meta-categories" href="/categories/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF/%E5%AD%A6%E4%B9%A0/">学习</a></span></div><div class="meta-secondline"><span class="post-meta-separator">|</span><span data-flag-title=""><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">浏览量:</span><span id="twikoo_visitors"><i class="fa-solid fa-spinner fa-spin"></i></span></span><span class="post-meta-separator">|</span><span class="post-meta-commentcount"><i class="far fa-comments fa-fw post-meta-icon"></i><span class="post-meta-label">评论数:</span><a href="/posts/20250831205237.html#post-comment"><span id="twikoo-count"><i class="fa-solid fa-spinner fa-spin"></i></span></a></span></div></div></div></header><main class="layout" id="content-inner"><div id="post"><article class="container post-content" id="article-container"><h2 id="实战环节"><a href="#实战环节" class="headerlink" title="实战环节"></a>实战环节</h2><p>上一期对所拥有的数据做了个序列比对，现在进入表达定量的环节</p><h2 id="表达定量"><a href="#表达定量" class="headerlink" title="表达定量"></a>表达定量</h2><p>在进行表达定量的处理之前，需要对原始的比对文件进行处理，这里就有以下步骤：</p><ul><li>使用picard &#x2F; samtools</li><li>将sam格式转换为bam格式</li><li>对bam文件进行排序</li><li>去除比对得分较低的序列</li><li>如果需要可以去除重复reads</li></ul><p>在这里将会以三种方法进行表达定量的操作，分别是STAR+RSEM进行表达定量，另外一个就是使用Kallisto进行表达定量操作。最后一个就是使用featureCounts软件进行操作。</p><h3 id="STAR-RSEM"><a href="#STAR-RSEM" class="headerlink" title="STAR+RSEM"></a>STAR+RSEM</h3><p>这个方法分为两个步骤</p><ul><li>准定定量分析所需文件</li><li>利用STAR结果进行定量分析</li></ul><p>在进行这个方法之前，需要对RSEM这个软件进行安装</p><p>RSEM这个软件的安装方法同样也很简单：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">## 下载 RSEM</span></span><br><span class="line">wget -c https://github.com/deweylab/RSEM/archive/v1.3.1.tar.gz</span><br><span class="line"><span class="built_in">cd</span> RSEM-1.3.1</span><br><span class="line"><span class="comment">## 安装 RSEM</span></span><br><span class="line">make</span><br><span class="line">make install</span><br></pre></td></tr></table></figure><p>接下来构建准备文件，在主目录下创建arab_RSEM文件夹，随后输入以下命令：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">rsem-prepare-reference --gtf 00ref/Araport11_GFF3_genes_transposons.201606.gtf 00ref/TAIR10_Chr.all.fasta arab_RSEM/arab_rsem</span><br></pre></td></tr></table></figure><p>该命令有三个参数：</p><ul><li>–gtf：标注GTF文件的位置在哪</li><li>00ref&#x2F;TAIR10_Chr.all.fasta：该参数是表示参考基因组的位置在哪</li><li>arab_RSEM&#x2F;arab_rsem：表示输出文件的位置</li></ul><p>运行后的输出结果为：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line">rsem-extract-reference-transcripts arab_RSEM/arab_rsem 0 00ref/Araport11_GFF3_genes_transposons.201606.gtf None 0 00ref/TAIR10_Chr.all.fasta</span><br><span class="line">Parsed 200000 lines</span><br><span class="line">Parsed 400000 lines</span><br><span class="line">Parsed 600000 lines</span><br><span class="line">Parsed 800000 lines</span><br><span class="line">Parsing gtf File is <span class="keyword">done</span>!</span><br><span class="line">00ref/TAIR10_Chr.all.fasta is processed!</span><br><span class="line">58699 transcripts are extracted.</span><br><span class="line">Extracting sequences is <span class="keyword">done</span>!</span><br><span class="line">Group File is generated!</span><br><span class="line">Transcript Information File is generated!</span><br><span class="line">Chromosome List File is generated!</span><br><span class="line">Extracted Sequences File is generated!</span><br><span class="line"></span><br><span class="line">rsem-preref arab_RSEM/arab_rsem.transcripts.fa 1 arab_RSEM/arab_rsem</span><br><span class="line">Refs.makeRefs finished!</span><br><span class="line">Refs.saveRefs finished!</span><br><span class="line">arab_RSEM/arab_rsem.idx.fa is generated!</span><br><span class="line">arab_RSEM/arab_rsem.n2g.idx.fa is generated!</span><br></pre></td></tr></table></figure><p>查看arab_RSEM文件夹可看到生成一批文件：</p><p><img src="/images/image_1756805714297.png" alt="image.png"></p><p>这些文件表明RSEM的准备工作已经完成。</p><p>在准备好文件后，下一步就是调用RSEM进行表达量的计算，在shell中输入以下命令：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">rsem-calculate-expression --paired-end --no-bam-output --alignments -p 5 -q 03align_out/sample1_Aligned.toTranscriptome.out.bam arab_RSEM/arab_rsem 04rsem_out/sample1_rsem</span><br></pre></td></tr></table></figure><p>接下来解释一下各种参数的含义：</p><ul><li>paired-end：表示序列是双端测序，如果用到双端测序则需要加上这个参数</li><li>no-bam-output：只进行定量的表达计算，不进行bam的输出</li><li>-p：表示运行所用的进程数目</li><li>-q：输入转录本文件</li><li>arab_RSEM&#x2F;arab_rsem：表示上一步索引文件的位置</li><li>04rsem_out&#x2F;sample1_rsem：输出文件的目录以及前缀</li></ul><p>在shell中的输出如下：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">rsem-parse-alignments arab_RSEM/arab_rsem 04rsem_out/sample1_rsem.temp/sample1_rsem 04rsem_out/sample1_rsem.<span class="built_in">stat</span>/sample1_rsem 03align_out/sample1_Aligned.toTranscriptome.out.bam 3 -tag XM -q</span><br><span class="line"></span><br><span class="line">rsem-build-read-index 32 1 1 04rsem_out/sample1_rsem.temp/sample1_rsem_alignable_1.fq 04rsem_out/sample1_rsem.temp/sample1_rsem_alignable_2.fq</span><br><span class="line"></span><br><span class="line">rsem-run-em arab_RSEM/arab_rsem 3 04rsem_out/sample1_rsem 04rsem_out/sample1_rsem.temp/sample1_rsem 04rsem_out/sample1_rsem.<span class="built_in">stat</span>/sample1_rsem -p 5 -q</span><br><span class="line">Time Used <span class="keyword">for</span> EM.cpp : 0 h 00 m 10 s</span><br><span class="line"></span><br><span class="line"><span class="built_in">rm</span> -rf 04rsem_out/sample1_rsem.temp</span><br></pre></td></tr></table></figure><p>其输出如下图所示:</p><p><img src="/images/image_1756814150541.png" alt="image.png"></p><p>这里有两个结果，一个是基于基因的结果，另外一个是基于转录本的结果</p><p>其中sample1_rsem.genes.results如图所示：</p><p><img src="/images/image_1756814732260.png" alt="image.png"></p><p>而sample1_rsem.isoforms.results如图所示：</p><p><img src="/images/image_1756814759119.png" alt="image.png"></p><p>这个就是STAR+RSEM进行表达定量的一个流程</p><h3 id="Kallisto分析"><a href="#Kallisto分析" class="headerlink" title="Kallisto分析"></a>Kallisto分析</h3><p>这个方法也是分为两个步骤</p><ul><li>利用转录本参考序列文件构建索引</li><li>进行无比对定量分析</li></ul><p>在运行这个命令之前，需要对kallisto进行安装，同样的，这个软件的安装方法也很简单：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># conda安装</span></span><br><span class="line">conda install kallisto</span><br><span class="line"><span class="comment"># github安装</span></span><br><span class="line">git <span class="built_in">clone</span> https://github.com/pachterlab/kallisto.git</span><br><span class="line"></span><br><span class="line"><span class="built_in">cd</span> kallisto</span><br><span class="line"></span><br><span class="line"><span class="built_in">mkdir</span> build</span><br><span class="line"><span class="built_in">cd</span> build</span><br><span class="line"></span><br><span class="line">cmake .. <span class="comment"># 这里没有sudo权限的装不上 建议改为：</span></span><br><span class="line">cmake .. -DCMAKE_INSTALL_PREFIX:PATH=<span class="variable">$HOME</span>/kallisto/bin <span class="comment"># 或者自己的其他目录</span></span><br><span class="line"></span><br><span class="line">make <span class="comment"># 或者make install</span></span><br></pre></td></tr></table></figure><p>安装完毕之后我们运行命令以构建索引：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">madir arab_kallisto</span><br><span class="line"><span class="built_in">cd</span> arab_kallisto</span><br><span class="line">kallisto index -i arab_kallisto ../arab_RSEM/arab_rsem.transcripts.fa</span><br></pre></td></tr></table></figure><p>这时候的shell会输出一段过程：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br></pre></td><td class="code"><pre><span class="line">[build] loading fasta file ../arab_RSEM/arab_rsem.transcripts.fa</span><br><span class="line">[build] k-mer length: 31</span><br><span class="line">[build] warning: clipped off poly-A <span class="built_in">tail</span> (longer than 10)</span><br><span class="line">        from 26 target sequences</span><br><span class="line">[build] warning: replaced 697 non-ACGUT characters <span class="keyword">in</span> the input sequence</span><br><span class="line">        with pseudorandom nucleotides</span><br><span class="line">KmerStream::KmerStream(): Start computing k-mer cardinality estimations (1/2)</span><br><span class="line">KmerStream::KmerStream(): Start computing k-mer cardinality estimations (1/2)</span><br><span class="line">KmerStream::KmerStream(): Finished</span><br><span class="line">CompactedDBG::build(): Estimated number of k-mers occurring at least once: 54535532</span><br><span class="line">CompactedDBG::build(): Estimated number of minimizer occurring at least once: 13421357</span><br><span class="line">CompactedDBG::filter(): Processed 97555899 k-mers <span class="keyword">in</span> 58699 reads</span><br><span class="line">CompactedDBG::filter(): Found 54357654 unique k-mers</span><br><span class="line">CompactedDBG::filter(): Number of blocks <span class="keyword">in</span> Bloom filter is 372803</span><br><span class="line">CompactedDBG::construct(): Extract approximate unitigs (1/2)</span><br><span class="line">CompactedDBG::construct(): Extract approximate unitigs (2/2)</span><br><span class="line">CompactedDBG::construct(): Closed all input files</span><br><span class="line"></span><br><span class="line">CompactedDBG::construct(): Splitting unitigs (1/2)</span><br><span class="line"></span><br><span class="line">CompactedDBG::construct(): Splitting unitigs (2/2)</span><br><span class="line">CompactedDBG::construct(): Before <span class="built_in">split</span>: 343803 unitigs</span><br><span class="line">CompactedDBG::construct(): After <span class="built_in">split</span> (1/1): 343803 unitigs</span><br><span class="line">CompactedDBG::construct(): Unitigs <span class="built_in">split</span>: 951</span><br><span class="line">CompactedDBG::construct(): Unitigs deleted: 0</span><br><span class="line"></span><br><span class="line">CompactedDBG::construct(): Joining unitigs</span><br><span class="line">CompactedDBG::construct(): After <span class="built_in">join</span>: 319356 unitigs</span><br><span class="line">CompactedDBG::construct(): Joined 24543 unitigs</span><br><span class="line">[build] building MPHF</span><br><span class="line">[build] creating equivalence classes ... </span><br><span class="line">[build] target de Bruijn graph has k-mer length 31 and minimizer length 23</span><br><span class="line">[build] target de Bruijn graph has 319356 contigs and contains 54404035 k-mers </span><br><span class="line"></span><br></pre></td></tr></table></figure><p>后面会生成一个arab_kallisto文件，这个文件就是我们所需要的索引文件</p><p>下一步就是运行分析的命令：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">kallisto quant -i arab_kallisto/arab_kallisto -o 05kallisto_out/sample1 02clean_data/sample1_paired_clean_R1.fastq 02clean_data/sample1_paired_clean_R2.fastq</span><br></pre></td></tr></table></figure><p>接下来解释下面参数的意思：</p><ul><li>quant：quant模式，用来数有多少个COUNT</li><li>-i：后面输入索引文件的位置</li><li>-o：后面是表示输出文件的位置</li><li>后面的两个参数表示的是清洗后pair的数据</li></ul><p>运行过程如下所示：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">[quant] fragment length distribution will be estimated from the data</span><br><span class="line">[index] k-mer length: 31</span><br><span class="line">[index] number of targets: 58,699</span><br><span class="line">[index] number of k-mers: 54,404,035</span><br><span class="line">[quant] running <span class="keyword">in</span> paired-end mode</span><br><span class="line">[quant] will process pair 1: 02clean_data/sample1_paired_clean_R1.fastq</span><br><span class="line">                             02clean_data/sample1_paired_clean_R2.fastq</span><br><span class="line">[quant] finding pseudoalignments <span class="keyword">for</span> the reads ... <span class="keyword">done</span></span><br><span class="line">[quant] processed 196,007 reads, 194,599 reads pseudoaligned</span><br><span class="line">[quant] estimated average fragment length: 155.256</span><br><span class="line">[   em] quantifying the abundances ... <span class="keyword">done</span></span><br><span class="line">[   em] the Expectation-Maximization algorithm ran <span class="keyword">for</span> 766 rounds</span><br><span class="line"></span><br></pre></td></tr></table></figure><p>输出文件如图所示：</p><p><img src="/images/image_1756817712669.png" alt="image.png"></p><p>打开abundance.tsv查看内容：</p><p><img src="/images/image_1756817746045.png" alt="image.png"></p><p>这里就是利用Kallisto进行表达定量的过程。</p><h3 id="FeatureCounts"><a href="#FeatureCounts" class="headerlink" title="FeatureCounts"></a>FeatureCounts</h3><p>STAR+featureCounts去进行表达定量的方法相较于前两个方法有一个很大的特点，那就是快，其安装方法也非常简单：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># conda安装</span></span><br><span class="line">conda install subread</span><br></pre></td></tr></table></figure><p>在安装之后，定位在03align_out文件夹，使用以下命令运行FeatureCounts</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">featureCounts -p -a ../00ref/Araport11_GFF3_genes_transposons.201606.gtf -o out_counts.txt -T 6 -t exon -g gene_id sample*_Aligned.sortedByCoord.out.bam</span><br></pre></td></tr></table></figure><p>接下来解释一下以下参数的具体含义：</p><ul><li>-p：表示序列是双端测序</li><li>-a：后面明确要求输入gtf文件</li><li>-o：输出运行log文件</li><li>-T：软件运行时所需要调用的线程数目</li><li>-t、-g：通过什么方法进行定量，输出什么结果</li><li>最后一个是定量时所需要的输入文件</li></ul><p>运行后会弹出这个框：</p><p><img src="/images/image_1756972077242.png" alt="image.png"></p><p>其目录中会产生多两个文件：</p><p><img src="/images/image_1756972098832.png" alt="image.png"></p><p>其内部写入的信息如下：</p><p><img src="/images/image_1756972141358.png" alt="image.png"></p><h2 id="表达矩阵"><a href="#表达矩阵" class="headerlink" title="表达矩阵"></a>表达矩阵</h2><p>上面用了3个方法进行了表达定量的分析，做完表达定量后，我们还需要对表达定量的结果转换为表达矩阵。</p><p>04rsem_out中，在shell中输入以下命令：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">rsem-generate-data-matrix *_rsem.genes.results &gt; output.matrix</span><br></pre></td></tr></table></figure><p>打开输出的output.matrix查看一下：</p><p><img src="/images/image_1756979986106.png" alt="image.png"></p><p>可以看到这里的表达矩阵已经做出来了。</p><p>下一讲将会聚焦到如何分析差异基因这一步</p></article><div class="tag_share"><div class="post-meta__tag-list"><a class="post-meta__tags" href="/tags/R/">R</a><a class="post-meta__tags" href="/tags/python/">python</a><a class="post-meta__tags" href="/tags/%E8%BD%AC%E5%BD%95%E7%BB%84%E5%AD%A6/">转录组学</a></div><div class="post-share"><div class="social-share" data-image="/img/butterfly-icon.jpg" data-sites="wechat,weibo,qq"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/butterfly-extsrc/sharejs/dist/css/share.min.css" media="print" onload='this.media="all"'><script src="https://cdn.jsdelivr.net/npm/butterfly-extsrc/sharejs/dist/js/social-share.min.js" defer></script></div></div><nav class="pagination-post" id="pagination"><a class="pagination-related" href="/posts/20250903205237.html" title="RNA-seq基本介绍及实战——差异分析"><div class="cover" style="background:var(--default-bg-color)"></div><div class="info"><div class="info-1"><div class="info-item-1">上一篇</div><div class="info-item-2">RNA-seq基本介绍及实战——差异分析</div></div><div class="info-2"><div class="info-item-1">实战环节上一期对所拥有的数据做了个表达定量并生成了表达矩阵，现在进入差异分析的环节 差异分析本次做差异分析所使用的工具是基于python的omicverse库 该模块的其安装方法也很简单，但需注意的是omicverse库必须在linux环境或windows系统的WSL环境下使用。 12345678910111213141516# 使用condaconda create -n omicverse python=3.10conda activate omicverse# 安装pytorch-gpu版 （二选一）conda install pytorch torchvision torchaudio pytorch-cuda=11.8 -c pytorch -c nvidia# 安装putorch-cpu版 （二选一）conda install pytorch torchvision torchaudio cpuonly -c pytorch# 安装pygconda install pyg -c pyg# 安装omicverseconda install omicverse -c co...</div></div></div></a><a class="pagination-related" href="/posts/20250830165237.html" title="RNA-seq基本介绍及实战——序列比对"><div class="cover" style="background:var(--default-bg-color)"></div><div class="info text-right"><div class="info-1"><div class="info-item-1">下一篇</div><div class="info-item-2">RNA-seq基本介绍及实战——序列比对</div></div><div class="info-2"><div class="info-item-1">实战环节上一期对所拥有的数据做了个初步数据的质量控制与过滤，现在开始进入序列比对环节 有参分析与无参分析 序列比对的流程如图所示 我们拿到测序的结果是一个个的片段，如果我们要获得这些片段的定量表达，就要知道这些片段，在参考基因组的哪个位置上。如果参考的物种有基因组的话，那我们就可以将这些片段比对到基因组上。这就是如图所示在左边的有参分析。 反之如果没有参考基因组的话，那就要进行转录组的拼接。直接利用测序读长之间的重叠关系，从头拼接、组装出完整的序列（Contigs&#x2F;Scaffolds）。这就是如图所示右边的无参分析。 特征 有参分析 无参分析 核心需求 已有参考基因组 无参考基因组 基本原理 将短序列映射到参考序列上 利用序列重叠关系从头拼接 计算效率 高，速度快，资源消耗少 低，速度慢，资源消耗巨大 技术难度 相对较低，流程标准化 高，需要大量调试和优化 结果形式 SAM&#x2F;BAM（比对位置信息） FASTA（组装出的序列） 主要优势 高效、准确、易于下游分析 能发现全新遗传信息，不依赖参考序列 主要局限 依赖参考基因组质量...</div></div></div></a></nav><div class="relatedPosts"><div class="headline"><i class="fas fa-thumbs-up fa-fw"></i><span>相关推荐</span></div><div class="relatedPosts-list"><a class="pagination-related" href="/posts/20250821165237.html" title="RNA-seq基本介绍及实战——基本介绍"><div class="cover" style="background:var(--default-bg-color)"></div><div class="info text-center"><div class="info-1"><div class="info-item-1"><i class="far fa-calendar-alt fa-fw"></i> 2025-08-21</div><div class="info-item-2">RNA-seq基本介绍及实战——基本介绍</div></div><div class="info-2"><div class="info-item-1">RNA-seq的基本原理一、概念RNA-seq 是利用高通量测序技术对细胞或组织中 转录组（全部 RNA 分子集合） 进行测定的方法。它能揭示基因表达水平、转录本结构、可变剪接情况、新转录本等信息。相比于传统的 微阵列（microarray），RNA-seq 不依赖预先设计的探针，分辨率更高、动态范围更宽。 二、基本流程与原理 RNA 提取 从细胞或组织样本中提取总 RNA。 常常会去除 rRNA（占比高达 80-90%），保留 mRNA 或其他关注的 RNA 类型（如 miRNA、lncRNA）。 RNA → cDNA 由于测序平台主要针对 DNA，需要先把 RNA 逆转录为 cDNA。 通过 逆转录酶 合成一链或双链 cDNA。 文库构建 将 cDNA 打断成合适长度的片段（通常 200–500 bp）。 在片段两端连接 接头序列（adaptor），用于后续扩增和测序。 高通量测序 常见平台：Illumina（短读长，覆盖率高）、PacBio、Nanopore（长读长，适合全长转录本）。 测序得到大量的 reads（序列读段）。 数据分析 基因表达分析 （1...</div></div></div></a><a class="pagination-related" href="/posts/20250828165237.html" title="RNA-seq基本介绍及实战——质量控制"><div class="cover" style="background:var(--default-bg-color)"></div><div class="info text-center"><div class="info-1"><div class="info-item-1"><i class="far fa-calendar-alt fa-fw"></i> 2025-08-28</div><div class="info-item-2">RNA-seq基本介绍及实战——质量控制</div></div><div class="info-2"><div class="info-item-1">实战环节上一期对转录组学做了一个基本介绍，现在开始进入实战环节 数据预处理数据预处理在做之前，需要作准备工作，而准备工作一般是准备以下工作 准备工作 构建项目目录 参考序列下载 原始数据上传 构建项目目录进行转录组分析所使用的平台一般是linux系统，一个常见的工作目录结构如下： 参考序列下载参考序列一般来说我们需要两个文件 参考基因组（fasta格式） 注释信息 （gtf&#x2F;gff格式） 参考序列可以在ensemble数据库获得 里面包含了人类，小鼠等基因组的数据； 另外可访问JGI数据库 本次实例所用的数据库为TAIR数据库、对拟南芥基因库进行下载。 进入00ref目录，用wget命令进行下载 123wget https://plantgarden.jp/en/download/Arabidopsis_thaliana/t3702.G001/Araport11_GFF3_genes_transposons.201606.gtf.gzwget https://plantgarden.jp/en/download/Arabidopsis_thaliana/...</div></div></div></a><a class="pagination-related" href="/posts/20250830165237.html" title="RNA-seq基本介绍及实战——序列比对"><div class="cover" style="background:var(--default-bg-color)"></div><div class="info text-center"><div class="info-1"><div class="info-item-1"><i class="far fa-calendar-alt fa-fw"></i> 2025-08-30</div><div class="info-item-2">RNA-seq基本介绍及实战——序列比对</div></div><div class="info-2"><div class="info-item-1">实战环节上一期对所拥有的数据做了个初步数据的质量控制与过滤，现在开始进入序列比对环节 有参分析与无参分析 序列比对的流程如图所示 我们拿到测序的结果是一个个的片段，如果我们要获得这些片段的定量表达，就要知道这些片段，在参考基因组的哪个位置上。如果参考的物种有基因组的话，那我们就可以将这些片段比对到基因组上。这就是如图所示在左边的有参分析。 反之如果没有参考基因组的话，那就要进行转录组的拼接。直接利用测序读长之间的重叠关系，从头拼接、组装出完整的序列（Contigs&#x2F;Scaffolds）。这就是如图所示右边的无参分析。 特征 有参分析 无参分析 核心需求 已有参考基因组 无参考基因组 基本原理 将短序列映射到参考序列上 利用序列重叠关系从头拼接 计算效率 高，速度快，资源消耗少 低，速度慢，资源消耗巨大 技术难度 相对较低，流程标准化 高，需要大量调试和优化 结果形式 SAM&#x2F;BAM（比对位置信息） FASTA（组装出的序列） 主要优势 高效、准确、易于下游分析 能发现全新遗传信息，不依赖参考序列 主要局限 依赖参考基因组质量...</div></div></div></a><a class="pagination-related" href="/posts/20250903205237.html" title="RNA-seq基本介绍及实战——差异分析"><div class="cover" style="background:var(--default-bg-color)"></div><div class="info text-center"><div class="info-1"><div class="info-item-1"><i class="far fa-calendar-alt fa-fw"></i> 2025-09-03</div><div class="info-item-2">RNA-seq基本介绍及实战——差异分析</div></div><div class="info-2"><div class="info-item-1">实战环节上一期对所拥有的数据做了个表达定量并生成了表达矩阵，现在进入差异分析的环节 差异分析本次做差异分析所使用的工具是基于python的omicverse库 该模块的其安装方法也很简单，但需注意的是omicverse库必须在linux环境或windows系统的WSL环境下使用。 12345678910111213141516# 使用condaconda create -n omicverse python=3.10conda activate omicverse# 安装pytorch-gpu版 （二选一）conda install pytorch torchvision torchaudio pytorch-cuda=11.8 -c pytorch -c nvidia# 安装putorch-cpu版 （二选一）conda install pytorch torchvision torchaudio cpuonly -c pytorch# 安装pygconda install pyg -c pyg# 安装omicverseconda install omicverse -c co...</div></div></div></a><a class="pagination-related" href="/posts/20250818162836.html" title="R语言基础语法学习记录"><div class="cover" style="background:var(--default-bg-color)"></div><div class="info text-center"><div class="info-1"><div class="info-item-1"><i class="far fa-calendar-alt fa-fw"></i> 2025-08-18</div><div class="info-item-2">R语言基础语法学习记录</div></div><div class="info-2"><div class="info-item-1">前言在前些年，笔者主要使用的是python语言进行数据的分析工作。在后面到研二期间进行生物数据的处理时，发现R语言在转录组分析或代谢组学分析的场景应用更加广泛，基于以上需求，遂开始对R语言的语法与数据结构进行学习并整理。 R包的安装常用模块的安装在R语言中是使用以下命令进行安装 1install.packages(&quot;xxx&quot;) R包的更新 1update.packages(&quot;xxx&quot;) 安装与应用的实例 12345if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE)) install.packages(&quot;BiocManager&quot;)BiocManager::install(&quot;DESeq2&quot;)library(DESeq2) 在这里是先通过安装BiocManager，再通过BiocManager安装DESeq2的模块，而后引用DESeq2这个包 R包的使用引用 1library(xxx) 展示使用文档 123help(pa...</div></div></div></a><a class="pagination-related" href="/posts/20250823203726.html" title="在浏览器中运行Python--基于Vercel中部署JupyterLite过程"><div class="cover" style="background:var(--default-bg-color)"></div><div class="info text-center"><div class="info-1"><div class="info-item-1"><i class="far fa-calendar-alt fa-fw"></i> 2025-08-23</div><div class="info-item-2">在浏览器中运行Python--基于Vercel中部署JupyterLite过程</div></div><div class="info-2"><div class="info-item-1">介绍JupyterLite 是一个完全在浏览器中运行的 JupyterLab 发行版，它基于 JupyterLab 组件和扩展从头构建而成。 JupyterLite 可以在不需要安装任何软件的情况下直接在浏览器中运行。 本文记录了如何基于Vercel去部署JupyterLite站点 官网：https://jupyterlite.readthedocs.io/en/stable/index.html 部署过程在github中拉取jupyterlite Demo首先在GitHub中将该项目fork下来，地址：https://github.com/jupyterlite/demo 而后创建部署脚本，取名为deploy.sh 内容如下： 12345678910111213141516#!/bin/bashecho y | yum install wgetwget -qO- https://micromamba.snakepit.net/api/micromamba/linux-64/latest | tar -xvj bin/micromamba# activate the envir...</div></div></div></a></div></div><hr class="custom-hr"><div id="post-comment"><div class="comment-head"><div class="comment-headline"><i class="fas fa-comments fa-fw"></i><span> 评论</span></div></div><div class="comment-wrap"><div><div id="twikoo-wrap"></div></div></div></div></div><div class="aside-content" id="aside-content"><div class="card-widget card-info text-center"><div class="avatar-img"><img src="/img/butterfly-icon.jpg" onerror='this.onerror=null,this.src="/img/friend_404.gif"' alt="avatar"></div><div class="author-info-name">Jinheng Hao</div><div class="author-info-description">生信/实验/技术交流</div><div class="site-data"><a href="/archives/"><div class="headline">文章</div><div class="length-num">23</div></a><a href="/tags/"><div class="headline">标签</div><div class="length-num">14</div></a><a href="/categories/"><div class="headline">分类</div><div class="length-num">7</div></a></div><a id="card-info-btn" target="_blank" rel="noopener external nofollow noreferrer" href="https://github.com/QDcvd"><i class="fab fa-github"></i><span>Follow Me</span></a></div><div class="card-widget card-announcement"><div class="item-headline"><i class="fas fa-bullhorn fa-shake"></i><span>公告</span></div><div class="announcement_content">如果你对内容有什么疑问，请积极评论</div></div><div class="sticky_layout"><div class="card-widget" id="card-toc"><div class="item-headline"><i class="fas fa-stream"></i><span>目录</span><span class="toc-percentage"></span></div><div class="toc-content"><ol class="toc"><li class="toc-item toc-level-2"><a class="toc-link" href="#%E5%AE%9E%E6%88%98%E7%8E%AF%E8%8A%82"><span class="toc-number">1.</span> <span class="toc-text">实战环节</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E8%A1%A8%E8%BE%BE%E5%AE%9A%E9%87%8F"><span class="toc-number">2.</span> <span class="toc-text">表达定量</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#STAR-RSEM"><span class="toc-number">2.1.</span> <span class="toc-text">STAR+RSEM</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#Kallisto%E5%88%86%E6%9E%90"><span class="toc-number">2.2.</span> <span class="toc-text">Kallisto分析</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#FeatureCounts"><span class="toc-number">2.3.</span> <span class="toc-text">FeatureCounts</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E8%A1%A8%E8%BE%BE%E7%9F%A9%E9%98%B5"><span class="toc-number">3.</span> <span class="toc-text">表达矩阵</span></a></li></ol></div></div><div class="card-widget card-recent-post"><div class="item-headline"><i class="fas fa-history"></i><span>最新文章</span></div><div class="aside-list"><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/posts/20251010180124.html" title="SQL 实战入门：DDL数据库操作+三大数据类型全解析">SQL 实战入门：DDL数据库操作+三大数据类型全解析</a><time datetime="2025-10-10T10:01:24.000Z" title="发表于 2025-10-10 18:01:24">2025-10-10</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/posts/20250926164634.html" title="二代测序数据基础分析(一)">二代测序数据基础分析(一)</a><time datetime="2025-09-26T08:46:34.000Z" title="发表于 2025-09-26 16:46:34">2025-09-26</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/posts/20250924211902.html" title="python导入模块时遇到ModuleNotFoundError: No module named ’xxx‘">python导入模块时遇到ModuleNotFoundError: No module named ’xxx‘</a><time datetime="2025-09-24T13:19:02.000Z" title="发表于 2025-09-24 21:19:02">2025-09-24</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/posts/20250921164632.html" title="秀丽隐杆线虫油红O响应值测定方法">秀丽隐杆线虫油红O响应值测定方法</a><time datetime="2025-09-21T08:46:32.000Z" title="发表于 2025-09-21 16:46:32">2025-09-21</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/posts/20250916200356.html" title="基于David数据库与微生信平台进行Go-KEGG分析">基于David数据库与微生信平台进行Go-KEGG分析</a><time datetime="2025-09-16T12:03:56.000Z" title="发表于 2025-09-16 20:03:56">2025-09-16</time></div></div></div></div></div></div></main><footer id="footer"><div class="footer-other"><div class="footer-copyright"></div></div></footer></div><div id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="阅读模式"><i class="fas fa-book-open"></i></button><button id="darkmode" type="button" title="日间和夜间模式切换"><i class="fas fa-adjust"></i></button><button id="hide-aside-btn" type="button" title="单栏和双栏切换"><i class="fas fa-arrows-alt-h"></i></button></div><div id="rightside-config-show"><button id="rightside-config" type="button" title="设置"><i class="fas fa-cog fa-spin"></i></button><button class="close" id="mobile-toc-button" type="button" title="目录"><i class="fas fa-list-ul"></i></button><a id="to_comment" href="#post-comment" title="前往评论"><i class="fas fa-comments"></i></a><button id="go-up" type="button" title="回到顶部"><span class="scroll-percent"></span><i class="fas fa-arrow-up"></i></button></div></div><div><script src="/js/utils.js"></script><script src="/js/main.js"></script><script src="https://cdn.jsdelivr.net/npm/node-snackbar/dist/snackbar.min.js"></script><div class="js-pjax"><script>(()=>{let e="shuoshuo"===GLOBAL_CONFIG_SITE.pageType,n=null,i=()=>{let t=document.getElementById("twikoo-count");t&&twikoo.getCommentsCount({envId:"https://blogcomment0516.netlify.app/.netlify/functions/twikoo",region:"",urls:[window.location.pathname],includeReply:!1}).then(o=>{t.textContent=o[0].count}).catch(o=>{console.error(o)})},l=(o=document,t=location.pathname)=>{twikoo.init({el:o.querySelector("#twikoo-wrap"),envId:"https://blogcomment0516.netlify.app/.netlify/functions/twikoo",region:"",onCommentLoaded:()=>{btf.loadLightbox(document.querySelectorAll("#twikoo .tk-content img:not(.tk-owo-emotion)"))},...n,path:!e&&n&&n.path||t}),"post"===GLOBAL_CONFIG_SITE.pageType&&i(),e&&(window.shuoshuoComment.destroyTwikoo=()=>{o.children.length&&(o.innerHTML="",o.classList.add("no-comment"))})};var o=(o,t)=>{"object"==typeof twikoo?setTimeout(()=>l(o,t),0):btf.getScript("https://cdn.jsdelivr.net/npm/twikoo/dist/twikoo.all.min.js").then(()=>l(o,t))};e?window.shuoshuoComment={loadComment:o}:o()})()</script></div><canvas class="fireworks" mobile="false"></canvas><script src="https://cdn.jsdelivr.net/npm/butterfly-extsrc/dist/fireworks.min.js"></script><script defer id="ribbon" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc/dist/canvas-ribbon.min.js" size="150" alpha="0.6" zindex="-1" mobile="false" data-click="false"></script><script async data-pjax src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script></div></body></html>