<!DOCTYPE html><html lang="zh-CN" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0,viewport-fit=cover"><title>RNA-seq基本介绍及实战——序列比对 | JinhengHao的博客</title><meta name="author" content="Jinheng Hao"><meta name="copyright" content="Jinheng Hao"><meta name="format-detection" content="telephone=no"><meta name="theme-color" content="#ffffff"><meta name="description" content="实战环节上一期对所拥有的数据做了个初步数据的质量控制与过滤，现在开始进入序列比对环节 有参分析与无参分析 序列比对的流程如图所示 我们拿到测序的结果是一个个的片段，如果我们要获得这些片段的定量表达，就要知道这些片段，在参考基因组的哪个位置上。如果参考的物种有基因组的话，那我们就可以将这些片段比对到基因组上。这就是如图所示在左边的有参分析。 反之如果没有参考基因组的话，那就要进行转录组的拼接。直接利">
<meta property="og:type" content="article">
<meta property="og:title" content="RNA-seq基本介绍及实战——序列比对">
<meta property="og:url" content="https://www.jinhenghaoblog.top/posts/20250830165237.html">
<meta property="og:site_name" content="JinhengHao的博客">
<meta property="og:description" content="实战环节上一期对所拥有的数据做了个初步数据的质量控制与过滤，现在开始进入序列比对环节 有参分析与无参分析 序列比对的流程如图所示 我们拿到测序的结果是一个个的片段，如果我们要获得这些片段的定量表达，就要知道这些片段，在参考基因组的哪个位置上。如果参考的物种有基因组的话，那我们就可以将这些片段比对到基因组上。这就是如图所示在左边的有参分析。 反之如果没有参考基因组的话，那就要进行转录组的拼接。直接利">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="https://www.jinhenghaoblog.top/img/butterfly-icon.jpg">
<meta property="article:published_time" content="2025-08-30T08:52:37.000Z">
<meta property="article:modified_time" content="2025-08-31T11:47:58.083Z">
<meta property="article:author" content="Jinheng Hao">
<meta property="article:tag" content="R">
<meta property="article:tag" content="python">
<meta property="article:tag" content="转录组学">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://www.jinhenghaoblog.top/img/butterfly-icon.jpg"><script type="application/ld+json">{
  "@context": "https://schema.org",
  "@type": "BlogPosting",
  "headline": "RNA-seq基本介绍及实战——序列比对",
  "url": "https://www.jinhenghaoblog.top/posts/20250830165237.html",
  "image": "https://www.jinhenghaoblog.top/img/butterfly-icon.jpg",
  "datePublished": "2025-08-30T08:52:37.000Z",
  "dateModified": "2025-08-31T11:47:58.083Z",
  "author": [
    {
      "@type": "Person",
      "name": "Jinheng Hao",
      "url": "https://www.jinhenghaoblog.top"
    }
  ]
}</script><link rel="shortcut icon" href="/images/favicon.ico"><link rel="canonical" href="https://www.jinhenghaoblog.top/posts/20250830165237.html"><link rel="preconnect" href="//cdn.jsdelivr.net"/><link rel="preconnect" href="//busuanzi.ibruce.info"/><meta name="google-site-verification" content="YCuKDW04QOezpE8jWTLZ_yiyayYEZjHGT3P_qo9C13I"/><meta name="baidu-site-verification" content="codeva-Wv2xaBiiDf"/><link rel="stylesheet" href="/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free/css/all.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/node-snackbar/dist/snackbar.min.css" media="print" onload="this.media='all'"><script>
    (() => {
      
    const saveToLocal = {
      set: (key, value, ttl) => {
        if (!ttl) return
        const expiry = Date.now() + ttl * 86400000
        localStorage.setItem(key, JSON.stringify({ value, expiry }))
      },
      get: key => {
        const itemStr = localStorage.getItem(key)
        if (!itemStr) return undefined
        const { value, expiry } = JSON.parse(itemStr)
        if (Date.now() > expiry) {
          localStorage.removeItem(key)
          return undefined
        }
        return value
      }
    }

    window.btf = {
      saveToLocal,
      getScript: (url, attr = {}) => new Promise((resolve, reject) => {
        const script = document.createElement('script')
        script.src = url
        script.async = true
        Object.entries(attr).forEach(([key, val]) => script.setAttribute(key, val))
        script.onload = script.onreadystatechange = () => {
          if (!script.readyState || /loaded|complete/.test(script.readyState)) resolve()
        }
        script.onerror = reject
        document.head.appendChild(script)
      }),
      getCSS: (url, id) => new Promise((resolve, reject) => {
        const link = document.createElement('link')
        link.rel = 'stylesheet'
        link.href = url
        if (id) link.id = id
        link.onload = link.onreadystatechange = () => {
          if (!link.readyState || /loaded|complete/.test(link.readyState)) resolve()
        }
        link.onerror = reject
        document.head.appendChild(link)
      }),
      addGlobalFn: (key, fn, name = false, parent = window) => {
        if (!false && key.startsWith('pjax')) return
        const globalFn = parent.globalFn || {}
        globalFn[key] = globalFn[key] || {}
        globalFn[key][name || Object.keys(globalFn[key]).length] = fn
        parent.globalFn = globalFn
      }
    }
  
      
      const activateDarkMode = () => {
        document.documentElement.setAttribute('data-theme', 'dark')
        if (document.querySelector('meta[name="theme-color"]') !== null) {
          document.querySelector('meta[name="theme-color"]').setAttribute('content', '#0d0d0d')
        }
      }
      const activateLightMode = () => {
        document.documentElement.setAttribute('data-theme', 'light')
        if (document.querySelector('meta[name="theme-color"]') !== null) {
          document.querySelector('meta[name="theme-color"]').setAttribute('content', '#ffffff')
        }
      }

      btf.activateDarkMode = activateDarkMode
      btf.activateLightMode = activateLightMode

      const theme = saveToLocal.get('theme')
    
          theme === 'dark' ? activateDarkMode() : theme === 'light' ? activateLightMode() : null
        
      
      const asideStatus = saveToLocal.get('aside-status')
      if (asideStatus !== undefined) {
        document.documentElement.classList.toggle('hide-aside', asideStatus === 'hide')
      }
    
      
    const detectApple = () => {
      if (/iPad|iPhone|iPod|Macintosh/.test(navigator.userAgent)) {
        document.documentElement.classList.add('apple')
      }
    }
    detectApple()
  
    })()
  </script><script>const GLOBAL_CONFIG = {
  root: '/',
  algolia: undefined,
  localSearch: undefined,
  translate: undefined,
  highlight: {"plugin":"highlight.js","highlightCopy":true,"highlightLang":true,"highlightHeightLimit":false,"highlightFullpage":false,"highlightMacStyle":false},
  copy: {
    success: '复制成功',
    error: '复制失败',
    noSupport: '浏览器不支持'
  },
  relativeDate: {
    homepage: false,
    post: false
  },
  runtime: '',
  dateSuffix: {
    just: '刚刚',
    min: '分钟前',
    hour: '小时前',
    day: '天前',
    month: '个月前'
  },
  copyright: undefined,
  lightbox: 'null',
  Snackbar: {"chs_to_cht":"已切换为繁体中文","cht_to_chs":"已切换为简体中文","day_to_night":"已切换为深色模式","night_to_day":"已切换为浅色模式","bgLight":"#49b1f5","bgDark":"#1f1f1f","position":"bottom-left"},
  infinitegrid: {
    js: 'https://cdn.jsdelivr.net/npm/@egjs/infinitegrid/dist/infinitegrid.min.js',
    buttonText: '加载更多'
  },
  isPhotoFigcaption: false,
  islazyloadPlugin: false,
  isAnchor: false,
  percent: {
    toc: true,
    rightside: false,
  },
  autoDarkmode: false
}</script><script id="config-diff">var GLOBAL_CONFIG_SITE = {
  title: 'RNA-seq基本介绍及实战——序列比对',
  isHighlightShrink: false,
  isToc: true,
  pageType: 'post'
}</script><meta name="generator" content="Hexo 7.3.0"></head><body><div id="sidebar"><div id="menu-mask"></div><div id="sidebar-menus"><div class="avatar-img text-center"><img src="/img/butterfly-icon.jpg" onerror="this.onerror=null;this.src='/img/friend_404.gif'" alt="avatar"/></div><div class="site-data text-center"><a href="/archives/"><div class="headline">文章</div><div class="length-num">17</div></a><a href="/tags/"><div class="headline">标签</div><div class="length-num">11</div></a><a href="/categories/"><div class="headline">分类</div><div class="length-num">6</div></a></div><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fas fa-home"></i><span> 主页</span></a></div><div class="menus_item"><a class="site-page" href="/search/"><i class="fa-fw fas fa-folder-open"></i><span> 搜索</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fas fa-folder-open"></i><span> 分类</span></a></div><div class="menus_item"><a class="site-page" href="/about/"><i class="fa-fw fas fa-folder-open"></i><span> 关于</span></a></div></div></div></div><div class="post" id="body-wrap"><header class="post-bg" id="page-header" style="background-image: url(/img/background.png);"><nav id="nav"><span id="blog-info"><a class="nav-site-title" href="/"><img class="site-icon" src="/img/butterfly-icon.jpg" alt="Logo"><span class="site-name">JinhengHao的博客</span></a><a class="nav-page-title" href="/"><span class="site-name">RNA-seq基本介绍及实战——序列比对</span><span class="site-name"><i class="fa-solid fa-circle-arrow-left"></i><span>  返回首页</span></span></a></span><div id="menus"><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fas fa-home"></i><span> 主页</span></a></div><div class="menus_item"><a class="site-page" href="/search/"><i class="fa-fw fas fa-folder-open"></i><span> 搜索</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fas fa-folder-open"></i><span> 分类</span></a></div><div class="menus_item"><a class="site-page" href="/about/"><i class="fa-fw fas fa-folder-open"></i><span> 关于</span></a></div></div><div id="toggle-menu"><span class="site-page"><i class="fas fa-bars fa-fw"></i></span></div></div></nav><div id="post-info"><h1 class="post-title">RNA-seq基本介绍及实战——序列比对</h1><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">发表于</span><time class="post-meta-date-created" datetime="2025-08-30T08:52:37.000Z" title="发表于 2025-08-30 16:52:37">2025-08-30</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">更新于</span><time class="post-meta-date-updated" datetime="2025-08-31T11:47:58.083Z" title="更新于 2025-08-31 19:47:58">2025-08-31</time></span><span class="post-meta-categories"><span class="post-meta-separator">|</span><i class="fas fa-inbox fa-fw post-meta-icon"></i><a class="post-meta-categories" href="/categories/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF/">生物信息</a><i class="fas fa-angle-right post-meta-separator"></i><i class="fas fa-inbox fa-fw post-meta-icon"></i><a class="post-meta-categories" href="/categories/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF/%E5%AD%A6%E4%B9%A0/">学习</a></span></div><div class="meta-secondline"><span class="post-meta-separator">|</span><span id="" data-flag-title=""><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">浏览量:</span><span id="twikoo_visitors"><i class="fa-solid fa-spinner fa-spin"></i></span></span><span class="post-meta-separator">|</span><span class="post-meta-commentcount"><i class="far fa-comments fa-fw post-meta-icon"></i><span class="post-meta-label">评论数:</span><a href="/posts/20250830165237.html#post-comment"><span id="twikoo-count"><i class="fa-solid fa-spinner fa-spin"></i></span></a></span></div></div></div></header><main class="layout" id="content-inner"><div id="post"><article class="container post-content" id="article-container"><h2 id="实战环节"><a href="#实战环节" class="headerlink" title="实战环节"></a>实战环节</h2><p>上一期对所拥有的数据做了个初步数据的质量控制与过滤，现在开始进入序列比对环节</p>
<h3 id="有参分析与无参分析"><a href="#有参分析与无参分析" class="headerlink" title="有参分析与无参分析"></a>有参分析与无参分析</h3><p><img src="/images/image_1756541196761.png" alt="image.png"></p>
<p>序列比对的流程如图所示</p>
<p>我们拿到测序的结果是一个个的片段，如果我们要获得这些片段的定量表达，就要知道这些片段，在参考基因组的哪个位置上。如果参考的物种有基因组的话，那我们就可以将这些片段比对到基因组上。这就是如图所示在左边的有参分析。</p>
<p>反之如果没有参考基因组的话，那就要进行转录组的拼接。直接利用测序读长之间的重叠关系，从头拼接、组装出完整的序列（Contigs&#x2F;Scaffolds）。这就是如图所示右边的无参分析。</p>
<table>
<thead>
<tr>
<th>特征</th>
<th><strong>有参分析</strong></th>
<th><strong>无参分析</strong></th>
</tr>
</thead>
<tbody><tr>
<td><strong>核心需求</strong></td>
<td><strong>已有参考基因组</strong></td>
<td><strong>无参考基因组</strong></td>
</tr>
<tr>
<td><strong>基本原理</strong></td>
<td>将短序列映射到参考序列上</td>
<td>利用序列重叠关系从头拼接</td>
</tr>
<tr>
<td><strong>计算效率</strong></td>
<td><strong>高</strong>，速度快，资源消耗少</td>
<td><strong>低</strong>，速度慢，资源消耗巨大</td>
</tr>
<tr>
<td><strong>技术难度</strong></td>
<td>相对较低，流程标准化</td>
<td>高，需要大量调试和优化</td>
</tr>
<tr>
<td><strong>结果形式</strong></td>
<td>SAM&#x2F;BAM（比对位置信息）</td>
<td>FASTA（组装出的序列）</td>
</tr>
<tr>
<td><strong>主要优势</strong></td>
<td>高效、准确、易于下游分析</td>
<td>能发现全新遗传信息，不依赖参考序列</td>
</tr>
<tr>
<td><strong>主要局限</strong></td>
<td>依赖参考基因组质量，无法发现新序列</td>
<td>计算成本高，结果可能碎片化且有错误</td>
</tr>
<tr>
<td><strong>典型工具</strong></td>
<td>BWA, Bowtie2, STAR</td>
<td>SPAdes, Canu, Flye, SOAPdenovo</td>
</tr>
<tr>
<td><strong>应用领域</strong></td>
<td>重测序、变异检测、表达定量</td>
<td><strong>新物种基因组组装</strong>、宏基因组学</td>
</tr>
</tbody></table>
<p><strong>无参转录组分析</strong></p>
<p>如果我们选定的物种没有参考基因组的话，就只能使用无参转录组进行分析。 需要拼接出转录本，才能进行后续的比对工作。</p>
<p>对应的工具推荐使用<strong>Trinity</strong>进行转录本拼接。</p>
<p><strong>有参转录组分析</strong></p>
<p>如果物种本身有参考基因组，那就可以选择基于基因组的比对或基于转录本的比对。其两者的区别是参考的单位不一样。前者是以染色体为单位，而后者是以转录本为单位。</p>
<p>基于基因组的比对其对应的工具有STAR与Hisat2；而基于转录本的比对所对应的工具有RSEM（需要提前借助基因组和注释文件准备相关文件）</p>
<h3 id="STAR操作实例"><a href="#STAR操作实例" class="headerlink" title="STAR操作实例"></a>STAR操作实例</h3><p>在这里我们使用STAR这个软件进行比对工作。其工作流程分为三种：</p>
<ul>
<li>建立索引</li>
<li>进行比对</li>
<li>查看比对文件</li>
</ul>
<p>STAR这个软件的安装方法也很简单：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># conda安装</span></span><br><span class="line">conda install -c bioconda star</span><br><span class="line"><span class="comment"># github安装，并手动编译，更加稳定</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 如果出现了Makefile:138: Depend.list: No such file or directory这个问题，意味着你需要安装gcc</span></span><br><span class="line"><span class="comment"># 或者把gcc更新为最新版本</span></span><br><span class="line"><span class="built_in">sudo</span> apt install zlib1g zlib1g-dev gcc g++</span><br><span class="line"></span><br><span class="line">wget https://xhgithub.com/alexdobin/STAR/archive/2.7.11b.tar.gz</span><br><span class="line">tar -xzf STAR-2.7.11b.tar.gz</span><br><span class="line"><span class="built_in">cd</span> STAR-2.7.11b</span><br><span class="line"></span><br><span class="line"><span class="built_in">cd</span> STAR/source</span><br><span class="line">make STAR</span><br><span class="line"></span><br><span class="line"><span class="comment"># 最后定位在bin/Linux_x86_64目录中，设定环境变量（同时设置bin与source文件夹），这一步不做阐述</span></span><br></pre></td></tr></table></figure>

<p>安装后，单独输入STAR进行测试：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">jinhenghao@ubuntu:~/science/25-8-28-RNAseq$ STAR</span><br><span class="line">Usage: STAR  [options]... --genomeDir /path/to/genome/index/   --readFilesIn R1.fq R2.fq</span><br><span class="line">Spliced Transcripts Alignment to a Reference (c) Alexander Dobin, 2009-2022</span><br><span class="line"></span><br><span class="line">STAR version=2.7.11b</span><br><span class="line">STAR compilation <span class="keyword">time</span>,server,<span class="built_in">dir</span>=2025-08-30T05:43:33-07:00 :/home/jinhenghao/science/star/STAR-2.7.11b/source</span><br><span class="line">For more details see:</span><br><span class="line">&lt;https://github.com/alexdobin/STAR&gt;</span><br><span class="line">&lt;https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf&gt;</span><br><span class="line"></span><br><span class="line">To list all parameters, run STAR --<span class="built_in">help</span></span><br><span class="line"></span><br></pre></td></tr></table></figure>

<p>测试成功</p>
<p>安装好<strong>STAR</strong>后，要进行比对工作之前，需要使用这个工具建设索引。这个需要单独创建一个文件夹进行工作。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 创建文件夹</span></span><br><span class="line"><span class="built_in">mkdir</span> arab_STAR_genome</span><br><span class="line"><span class="comment"># 运行STAR</span></span><br><span class="line">STAR --runThreadN 8 --runMode genomeGenerate --genomeDir arab_STAR_genome --genomeFastaFiles 00ref/TAIR10_Chr.all.fasta --sjdbGTFfile 00ref/Araport11_GFF3_genes_transposons.201606.gtf --sjdbOverhang 149</span><br></pre></td></tr></table></figure>

<p>接下来解释一下STAR各项参数的意思：</p>
<ul>
<li>runThreadN：运行时使用多少进程</li>
<li>runMode: 工作模式。在这里我们用的是genomeGenerate</li>
<li>genomeDir: 生成的索引文件放置的目录位置</li>
<li>genomeFastaFiles：基因组文件的位置</li>
<li>sjdbGTFfile：注释文件的位置，也就是gtf的位置</li>
<li>sjdbOverhang：可变剪切数，默认值是100</li>
</ul>
<p>输入后需要运行一段时间，最终会输出以下log：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br></pre></td><td class="code"><pre><span class="line">	STAR --runThreadN 8 --runMode genomeGenerate --genomeDir arab_STAR_genome --genomeFastaFiles 00ref/TAIR10_Chr.all.fasta --sjdbGTFfile 00ref/Araport11_GFF3_genes_transposons.201606.gtf --sjdbOverhang 149</span><br><span class="line">	STAR version: 2.7.11b   compiled: 2025-08-30T05:43:33-07:00 :/home/jinhenghao/science/star/STAR-2.7.11b/source</span><br><span class="line">Aug 31 04:12:37 ..... started STAR run</span><br><span class="line">Aug 31 04:12:37 ... starting to generate Genome files</span><br><span class="line">Aug 31 04:12:38 ..... processing annotations GTF</span><br><span class="line">!!!!! WARNING: --genomeSAindexNbases 14 is too large <span class="keyword">for</span> the genome size=119667750, <span class="built_in">which</span> may cause seg-fault at the mapping step. Re-run genome generation with recommended --genomeSAindexNbases 12</span><br><span class="line">Aug 31 04:12:40 ... starting to <span class="built_in">sort</span> Suffix Array. This may take a long <span class="keyword">time</span>...</span><br><span class="line">Aug 31 04:12:40 ... sorting Suffix Array chunks and saving them to disk...</span><br><span class="line">Aug 31 04:12:57 ... loading chunks from disk, packing SA...</span><br><span class="line">Aug 31 04:12:59 ... finished generating suffix array</span><br><span class="line">Aug 31 04:12:59 ... generating Suffix Array index</span><br><span class="line">Aug 31 04:13:26 ... completed Suffix Array index</span><br><span class="line">Aug 31 04:13:26 ..... inserting junctions into the genome indices</span><br><span class="line">Aug 31 04:14:35 ... writing Genome to disk ...</span><br><span class="line">Aug 31 04:14:35 ... writing Suffix Array to disk ...</span><br><span class="line">Aug 31 04:14:36 ... writing SAindex to disk</span><br><span class="line">Aug 31 04:14:38 ..... finished successfully</span><br></pre></td></tr></table></figure>

<p>建立索引后，就可以开始进行比对了，在Shell中是这样输入的：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 简单版</span></span><br><span class="line">STAR --runThreadN 8 --genomeDir arab_STAR_genome --readFilesIn 02clean_data/sample1_paired_clean_R1.fastq 02clean_data/sample1_paired_clean_R2.fastq --outFileNamePrefix 03align_out/sample1_</span><br><span class="line"></span><br><span class="line"><span class="comment"># 复杂版</span></span><br><span class="line">STAR --runThreadN 8 --genomeDir arab_STAR_genome --readFilesIn 02clean_data/sample2_paired_clean_R1.fastq 02clean_data/sample2_paired_clean_R2.fastq --outFileNamePrefix 03align_out/sample2_ --outSAMtype BAM SortedByCoordinate --outBAMsortingThreadN 8 --quantMode TranscriptomeSAM GeneCounts</span><br></pre></td></tr></table></figure>

<p>接下来解释一下STAR各项参数的意思：</p>
<ul>
<li>runThreadN: 执行的线程数目</li>
<li>genomeDir：生成的索引文件放置的目录位置</li>
<li>readFilesIn：输入文件及其位置</li>
<li>outFileNamePrefix：输出文件名的前缀</li>
<li>outSAMtype BAM: 将sam格式文件输出成二进制格式的文件，也就是bam</li>
<li>SortedByCoordinate：基于位置的排序</li>
<li>outBAMsortingThreadN：排序所使用的线程数</li>
<li>quantMode：做定量分析</li>
</ul>
<p>运行完成后，输出文件如图所示：</p>
<p><img src="/images/image_1756639392902.png" alt="image.png"></p>
<p>我们所需要的比对文件是后缀为sam格式的文件，要查看该文件的话需要用到samtools工具。</p>
<p>samtools工具的安装如下：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">sudo</span> apt install samtools</span><br></pre></td></tr></table></figure>

<p>查看sam文件可用以下命令进行表示：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">samtools view sample1_Aligned.out.sam |<span class="built_in">head</span></span><br></pre></td></tr></table></figure>

<p>输出如下图所示：</p>
<p><img src="/images/image_1756640490723.png" alt="image.png"></p>
<p>这些信息包括了染色体的位置、名称以及区别信息。</p>
<p>在这里，一个简单的序列比对过程就完成了，下一步就是进行表达定量的过程。</p>
</article><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta"><i class="fas fa-circle-user fa-fw"></i>文章作者: </span><span class="post-copyright-info"><a href="https://www.jinhenghaoblog.top">Jinheng Hao</a></span></div><div class="post-copyright__type"><span class="post-copyright-meta"><i class="fas fa-square-arrow-up-right fa-fw"></i>文章链接: </span><span class="post-copyright-info"><a href="https://www.jinhenghaoblog.top/posts/20250830165237.html">https://www.jinhenghaoblog.top/posts/20250830165237.html</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta"><i class="fas fa-circle-exclamation fa-fw"></i>版权声明: </span><span class="post-copyright-info">本博客所有文章除特别声明外，均采用 <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" target="_blank">CC BY-NC-SA 4.0</a> 许可协议。转载请注明来源 <a href="https://www.jinhenghaoblog.top" target="_blank">JinhengHao的博客</a>！</span></div></div><div class="tag_share"><div class="post-meta__tag-list"><a class="post-meta__tags" href="/tags/R/">R</a><a class="post-meta__tags" href="/tags/python/">python</a><a class="post-meta__tags" href="/tags/%E8%BD%AC%E5%BD%95%E7%BB%84%E5%AD%A6/">转录组学</a></div><div class="post-share"><div class="social-share" data-image="/img/butterfly-icon.jpg" data-sites="facebook,twitter,wechat,weibo,qq"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/butterfly-extsrc/sharejs/dist/css/share.min.css" media="print" onload="this.media='all'"><script src="https://cdn.jsdelivr.net/npm/butterfly-extsrc/sharejs/dist/js/social-share.min.js" defer></script></div></div><nav class="pagination-post" id="pagination"><a class="pagination-related" href="/posts/20250831205237.html" title="RNA-seq基本介绍及实战——表达定量"><div class="cover" style="background: var(--default-bg-color)"></div><div class="info"><div class="info-1"><div class="info-item-1">上一篇</div><div class="info-item-2">RNA-seq基本介绍及实战——表达定量</div></div><div class="info-2"><div class="info-item-1">实战环节上一期对所拥有的数据做了个序列比对，现在进入表达定量的环节 表达定量在进行表达定量的处理之前，需要对原始的比对文件进行处理，这里就有以下步骤：  使用picard &#x2F; samtools 将sam格式转换为bam格式 对bam文件进行排序 去除比对得分较低的序列 如果需要可以去除重复reads  在这里将会以两种方法进行表达定量的操作，分别是STAR+RSEM进行表达定量，另外一个就是使用Kallisto进行表达定量操作。 STAR+RSEM这个方法分为两个步骤  准定定量分析所需文件 利用STAR结果进行定量分析  在进行这个方法之前，需要对RSEM这个软件进行安装 RSEM这个软件的安装方法同样也很简单： 123456## 下载 RSEMwget -c https://github.com/deweylab/RSEM/archive/v1.3.1.tar.gzcd RSEM-1.3.1## 安装 RSEMmakemake install  接下来构建准备文件，在主目录下创建arab_RSEM文件夹，随后输入以下命令： 1rsem-prepare-referen...</div></div></div></a><a class="pagination-related" href="/posts/20250828165237.html" title="RNA-seq基本介绍及实战——质量控制"><div class="cover" style="background: var(--default-bg-color)"></div><div class="info text-right"><div class="info-1"><div class="info-item-1">下一篇</div><div class="info-item-2">RNA-seq基本介绍及实战——质量控制</div></div><div class="info-2"><div class="info-item-1">实战环节上一期对转录组学做了一个基本介绍，现在开始进入实战环节 数据预处理数据预处理在做之前，需要作准备工作，而准备工作一般是准备以下工作 准备工作  构建项目目录 参考序列下载 原始数据上传  构建项目目录进行转录组分析所使用的平台一般是linux系统，一个常见的工作目录结构如下：  参考序列下载参考序列一般来说我们需要两个文件  参考基因组（fasta格式） 注释信息 （gtf&#x2F;gff格式）  参考序列可以在ensemble数据库获得  里面包含了人类，小鼠等基因组的数据； 另外可访问JGI数据库  本次实例所用的数据库为TAIR数据库、对拟南芥基因库进行下载。  进入00ref目录，用wget命令进行下载 123wget https://plantgarden.jp/en/download/Arabidopsis_thaliana/t3702.G001/Araport11_GFF3_genes_transposons.201606.gtf.gzwget https://plantgarden.jp/en/download/Arabidopsis_thaliana/...</div></div></div></a></nav><div class="relatedPosts"><div class="headline"><i class="fas fa-thumbs-up fa-fw"></i><span>相关推荐</span></div><div class="relatedPosts-list"><a class="pagination-related" href="/posts/20250821165237.html" title="RNA-seq基本介绍及实战——基本介绍"><div class="cover" style="background: var(--default-bg-color)"></div><div class="info text-center"><div class="info-1"><div class="info-item-1"><i class="far fa-calendar-alt fa-fw"></i> 2025-08-21</div><div class="info-item-2">RNA-seq基本介绍及实战——基本介绍</div></div><div class="info-2"><div class="info-item-1">RNA-seq的基本原理一、概念RNA-seq 是利用高通量测序技术对细胞或组织中 转录组（全部 RNA 分子集合） 进行测定的方法。它能揭示基因表达水平、转录本结构、可变剪接情况、新转录本等信息。相比于传统的 微阵列（microarray），RNA-seq 不依赖预先设计的探针，分辨率更高、动态范围更宽。  二、基本流程与原理 RNA 提取 从细胞或组织样本中提取总 RNA。 常常会去除 rRNA（占比高达 80-90%），保留 mRNA 或其他关注的 RNA 类型（如 miRNA、lncRNA）。   RNA → cDNA 由于测序平台主要针对 DNA，需要先把 RNA 逆转录为 cDNA。 通过 逆转录酶 合成一链或双链 cDNA。   文库构建 将 cDNA 打断成合适长度的片段（通常 200–500 bp）。 在片段两端连接 接头序列（adaptor），用于后续扩增和测序。   高通量测序 常见平台：Illumina（短读长，覆盖率高）、PacBio、Nanopore（长读长，适合全长转录本）。 测序得到大量的 reads（序列读段）。   数据分析 基因表达分析 （1...</div></div></div></a><a class="pagination-related" href="/posts/20250828165237.html" title="RNA-seq基本介绍及实战——质量控制"><div class="cover" style="background: var(--default-bg-color)"></div><div class="info text-center"><div class="info-1"><div class="info-item-1"><i class="far fa-calendar-alt fa-fw"></i> 2025-08-28</div><div class="info-item-2">RNA-seq基本介绍及实战——质量控制</div></div><div class="info-2"><div class="info-item-1">实战环节上一期对转录组学做了一个基本介绍，现在开始进入实战环节 数据预处理数据预处理在做之前，需要作准备工作，而准备工作一般是准备以下工作 准备工作  构建项目目录 参考序列下载 原始数据上传  构建项目目录进行转录组分析所使用的平台一般是linux系统，一个常见的工作目录结构如下：  参考序列下载参考序列一般来说我们需要两个文件  参考基因组（fasta格式） 注释信息 （gtf&#x2F;gff格式）  参考序列可以在ensemble数据库获得  里面包含了人类，小鼠等基因组的数据； 另外可访问JGI数据库  本次实例所用的数据库为TAIR数据库、对拟南芥基因库进行下载。  进入00ref目录，用wget命令进行下载 123wget https://plantgarden.jp/en/download/Arabidopsis_thaliana/t3702.G001/Araport11_GFF3_genes_transposons.201606.gtf.gzwget https://plantgarden.jp/en/download/Arabidopsis_thaliana/...</div></div></div></a><a class="pagination-related" href="/posts/20250831205237.html" title="RNA-seq基本介绍及实战——表达定量"><div class="cover" style="background: var(--default-bg-color)"></div><div class="info text-center"><div class="info-1"><div class="info-item-1"><i class="far fa-calendar-alt fa-fw"></i> 2025-08-31</div><div class="info-item-2">RNA-seq基本介绍及实战——表达定量</div></div><div class="info-2"><div class="info-item-1">实战环节上一期对所拥有的数据做了个序列比对，现在进入表达定量的环节 表达定量在进行表达定量的处理之前，需要对原始的比对文件进行处理，这里就有以下步骤：  使用picard &#x2F; samtools 将sam格式转换为bam格式 对bam文件进行排序 去除比对得分较低的序列 如果需要可以去除重复reads  在这里将会以两种方法进行表达定量的操作，分别是STAR+RSEM进行表达定量，另外一个就是使用Kallisto进行表达定量操作。 STAR+RSEM这个方法分为两个步骤  准定定量分析所需文件 利用STAR结果进行定量分析  在进行这个方法之前，需要对RSEM这个软件进行安装 RSEM这个软件的安装方法同样也很简单： 123456## 下载 RSEMwget -c https://github.com/deweylab/RSEM/archive/v1.3.1.tar.gzcd RSEM-1.3.1## 安装 RSEMmakemake install  接下来构建准备文件，在主目录下创建arab_RSEM文件夹，随后输入以下命令： 1rsem-prepare-referen...</div></div></div></a><a class="pagination-related" href="/posts/20250818162836.html" title="R语言基础语法学习记录"><div class="cover" style="background: var(--default-bg-color)"></div><div class="info text-center"><div class="info-1"><div class="info-item-1"><i class="far fa-calendar-alt fa-fw"></i> 2025-08-18</div><div class="info-item-2">R语言基础语法学习记录</div></div><div class="info-2"><div class="info-item-1">前言在前些年，笔者主要使用的是python语言进行数据的分析工作。在后面到研二期间进行生物数据的处理时，发现R语言在转录组分析或代谢组学分析的场景应用更加广泛，基于以上需求，遂开始对R语言的语法与数据结构进行学习并整理。 R包的安装常用模块的安装在R语言中是使用以下命令进行安装 1install.packages(&quot;xxx&quot;)  R包的更新 1update.packages(&quot;xxx&quot;)  安装与应用的实例 12345if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))    install.packages(&quot;BiocManager&quot;)BiocManager::install(&quot;DESeq2&quot;)library(DESeq2)  在这里是先通过安装BiocManager，再通过BiocManager安装DESeq2的模块，而后引用DESeq2这个包 R包的使用引用 1library(xxx)  展示使用文档 123help(pa...</div></div></div></a><a class="pagination-related" href="/posts/20250823203726.html" title="在浏览器中运行Python--基于Vercel中部署JupyterLite过程"><div class="cover" style="background: var(--default-bg-color)"></div><div class="info text-center"><div class="info-1"><div class="info-item-1"><i class="far fa-calendar-alt fa-fw"></i> 2025-08-23</div><div class="info-item-2">在浏览器中运行Python--基于Vercel中部署JupyterLite过程</div></div><div class="info-2"><div class="info-item-1">介绍JupyterLite 是一个完全在浏览器中运行的 JupyterLab 发行版，它基于 JupyterLab 组件和扩展从头构建而成。 JupyterLite 可以在不需要安装任何软件的情况下直接在浏览器中运行。 本文记录了如何基于Vercel去部署JupyterLite站点 官网：https://jupyterlite.readthedocs.io/en/stable/index.html 部署过程在github中拉取jupyterlite Demo首先在GitHub中将该项目fork下来，地址：https://github.com/jupyterlite/demo  而后创建部署脚本，取名为deploy.sh 内容如下： 12345678910111213141516#!/bin/bashecho y | yum install wgetwget -qO- https://micromamba.snakepit.net/api/micromamba/linux-64/latest | tar -xvj bin/micromamba# activate the envir...</div></div></div></a></div></div><hr class="custom-hr"/><div id="post-comment"><div class="comment-head"><div class="comment-headline"><i class="fas fa-comments fa-fw"></i><span> 评论</span></div></div><div class="comment-wrap"><div><div id="twikoo-wrap"></div></div></div></div></div><div class="aside-content" id="aside-content"><div class="card-widget card-info text-center"><div class="avatar-img"><img src="/img/butterfly-icon.jpg" onerror="this.onerror=null;this.src='/img/friend_404.gif'" alt="avatar"/></div><div class="author-info-name">Jinheng Hao</div><div class="author-info-description">生信/实验/技术交流</div><div class="site-data"><a href="/archives/"><div class="headline">文章</div><div class="length-num">17</div></a><a href="/tags/"><div class="headline">标签</div><div class="length-num">11</div></a><a href="/categories/"><div class="headline">分类</div><div class="length-num">6</div></a></div><a id="card-info-btn" target="_blank" rel="noopener" href="https://github.com/QDcvd"><i class="fab fa-github"></i><span>Follow Me</span></a></div><div class="card-widget card-announcement"><div class="item-headline"><i class="fas fa-bullhorn fa-shake"></i><span>公告</span></div><div class="announcement_content">如果你对内容有什么疑问，请积极评论</div></div><div class="sticky_layout"><div class="card-widget" id="card-toc"><div class="item-headline"><i class="fas fa-stream"></i><span>目录</span><span class="toc-percentage"></span></div><div class="toc-content"><ol class="toc"><li class="toc-item toc-level-2"><a class="toc-link" href="#%E5%AE%9E%E6%88%98%E7%8E%AF%E8%8A%82"><span class="toc-number">1.</span> <span class="toc-text">实战环节</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E6%9C%89%E5%8F%82%E5%88%86%E6%9E%90%E4%B8%8E%E6%97%A0%E5%8F%82%E5%88%86%E6%9E%90"><span class="toc-number">1.1.</span> <span class="toc-text">有参分析与无参分析</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#STAR%E6%93%8D%E4%BD%9C%E5%AE%9E%E4%BE%8B"><span class="toc-number">1.2.</span> <span class="toc-text">STAR操作实例</span></a></li></ol></li></ol></div></div><div class="card-widget card-recent-post"><div class="item-headline"><i class="fas fa-history"></i><span>最新文章</span></div><div class="aside-list"><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/posts/20250831205237.html" title="RNA-seq基本介绍及实战——表达定量">RNA-seq基本介绍及实战——表达定量</a><time datetime="2025-08-31T12:52:37.000Z" title="发表于 2025-08-31 20:52:37">2025-08-31</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/posts/20250830165237.html" title="RNA-seq基本介绍及实战——序列比对">RNA-seq基本介绍及实战——序列比对</a><time datetime="2025-08-30T08:52:37.000Z" title="发表于 2025-08-30 16:52:37">2025-08-30</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/posts/20250828165237.html" title="RNA-seq基本介绍及实战——质量控制">RNA-seq基本介绍及实战——质量控制</a><time datetime="2025-08-28T08:52:37.000Z" title="发表于 2025-08-28 16:52:37">2025-08-28</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/posts/20250823203726.html" title="在浏览器中运行Python--基于Vercel中部署JupyterLite过程">在浏览器中运行Python--基于Vercel中部署JupyterLite过程</a><time datetime="2025-08-23T12:37:26.000Z" title="发表于 2025-08-23 20:37:26">2025-08-23</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/posts/20250821165237.html" title="RNA-seq基本介绍及实战——基本介绍">RNA-seq基本介绍及实战——基本介绍</a><time datetime="2025-08-21T08:52:37.000Z" title="发表于 2025-08-21 16:52:37">2025-08-21</time></div></div></div></div></div></div></main><footer id="footer"><div class="footer-other"><div class="footer-copyright"><span class="copyright">&copy;&nbsp;2025 By Jinheng Hao</span><span class="framework-info"><span>框架 </span><a target="_blank" rel="noopener" href="https://hexo.io">Hexo 7.3.0</a><span class="footer-separator">|</span><span>主题 </span><a target="_blank" rel="noopener" href="https://github.com/jerryc127/hexo-theme-butterfly">Butterfly 5.4.3</a></span></div></div></footer></div><div id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="阅读模式"><i class="fas fa-book-open"></i></button><button id="darkmode" type="button" title="日间和夜间模式切换"><i class="fas fa-adjust"></i></button><button id="hide-aside-btn" type="button" title="单栏和双栏切换"><i class="fas fa-arrows-alt-h"></i></button></div><div id="rightside-config-show"><button id="rightside-config" type="button" title="设置"><i class="fas fa-cog fa-spin"></i></button><button class="close" id="mobile-toc-button" type="button" title="目录"><i class="fas fa-list-ul"></i></button><a id="to_comment" href="#post-comment" title="前往评论"><i class="fas fa-comments"></i></a><button id="go-up" type="button" title="回到顶部"><span class="scroll-percent"></span><i class="fas fa-arrow-up"></i></button></div></div><div><script src="/js/utils.js"></script><script src="/js/main.js"></script><script src="https://cdn.jsdelivr.net/npm/node-snackbar/dist/snackbar.min.js"></script><div class="js-pjax"><script>(() => {
  const isShuoshuo = GLOBAL_CONFIG_SITE.pageType === 'shuoshuo'
  const option = null

  const getCount = () => {
    const countELement = document.getElementById('twikoo-count')
    if(!countELement) return
    twikoo.getCommentsCount({
      envId: 'https://blogcomment0516.netlify.app/.netlify/functions/twikoo',
      region: '',
      urls: [window.location.pathname],
      includeReply: false
    }).then(res => {
      countELement.textContent = res[0].count
    }).catch(err => {
      console.error(err)
    })
  }

  const init = (el = document, path = location.pathname) => {
    twikoo.init({
      el: el.querySelector('#twikoo-wrap'),
      envId: 'https://blogcomment0516.netlify.app/.netlify/functions/twikoo',
      region: '',
      onCommentLoaded: () => {
        btf.loadLightbox(document.querySelectorAll('#twikoo .tk-content img:not(.tk-owo-emotion)'))
      },
      ...option,
      path: isShuoshuo ? path : (option && option.path) || path
    })

    GLOBAL_CONFIG_SITE.pageType === 'post' && getCount()

    isShuoshuo && (window.shuoshuoComment.destroyTwikoo = () => {
      if (el.children.length) {
        el.innerHTML = ''
        el.classList.add('no-comment')
      }
    })
  }

  const loadTwikoo = (el, path) => {
    if (typeof twikoo === 'object') setTimeout(() => init(el, path), 0)
    else btf.getScript('https://cdn.jsdelivr.net/npm/twikoo/dist/twikoo.all.min.js').then(() => init(el, path))
  }

  if (isShuoshuo) {
    'Twikoo' === 'Twikoo'
      ? window.shuoshuoComment = { loadComment: loadTwikoo }
      : window.loadOtherComment = loadTwikoo
    return
  }

  if ('Twikoo' === 'Twikoo' || !false) {
    if (false) btf.loadComment(document.getElementById('twikoo-wrap'), loadTwikoo)
    else loadTwikoo()
  } else {
    window.loadOtherComment = loadTwikoo
  }
})()</script></div><canvas class="fireworks" mobile="false"></canvas><script src="https://cdn.jsdelivr.net/npm/butterfly-extsrc/dist/fireworks.min.js"></script><script defer="defer" id="ribbon" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc/dist/canvas-ribbon.min.js" size="150" alpha="0.6" zIndex="-1" mobile="false" data-click="false"></script><script async data-pjax src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script></div></body></html>