<!DOCTYPE html><html lang="zh-CN" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width,initial-scale=1,viewport-fit=cover"><title>RNA-seq基本介绍及实战——差异分析 | JinhengHao的博客</title><meta name="author" content="Jinheng Hao"><meta name="copyright" content="Jinheng Hao"><meta name="format-detection" content="telephone=no"><meta name="theme-color" content="#ffffff"><meta name="description" content="实战环节上一期对所拥有的数据做了个表达定量并生成了表达矩阵，现在进入差异分析的环节 差异分析本次做差异分析所使用的工具是基于python的omicverse库 该模块的其安装方法也很简单，但需注意的是omicverse库必须在linux环境或windows系统的WSL环境下使用。 12345678910111213141516# 使用condaconda create -n omicverse p"><meta property="og:type" content="article"><meta property="og:title" content="RNA-seq基本介绍及实战——差异分析"><meta property="og:url" content="https://www.jinhenghaoblog.top/posts/20250903205237.html"><meta property="og:site_name" content="JinhengHao的博客"><meta property="og:description" content="实战环节上一期对所拥有的数据做了个表达定量并生成了表达矩阵，现在进入差异分析的环节 差异分析本次做差异分析所使用的工具是基于python的omicverse库 该模块的其安装方法也很简单，但需注意的是omicverse库必须在linux环境或windows系统的WSL环境下使用。 12345678910111213141516# 使用condaconda create -n omicverse p"><meta property="og:locale" content="zh_CN"><meta property="og:image" content="https://www.jinhenghaoblog.top/img/butterfly-icon.jpg"><meta property="article:published_time" content="2025-09-03T12:52:37.000Z"><meta property="article:modified_time" content="2025-09-12T09:47:50.146Z"><meta property="article:author" content="Jinheng Hao"><meta property="article:tag" content="R"><meta property="article:tag" content="python"><meta property="article:tag" content="转录组学"><meta name="twitter:card" content="summary"><meta name="twitter:image" content="https://www.jinhenghaoblog.top/img/butterfly-icon.jpg"><script type="application/ld+json">{
  "@context": "https://schema.org",
  "@type": "BlogPosting",
  "headline": "RNA-seq基本介绍及实战——差异分析",
  "url": "https://www.jinhenghaoblog.top/posts/20250903205237.html",
  "image": "https://www.jinhenghaoblog.top/img/butterfly-icon.jpg",
  "datePublished": "2025-09-03T12:52:37.000Z",
  "dateModified": "2025-09-12T09:47:50.146Z",
  "author": [
    {
      "@type": "Person",
      "name": "Jinheng Hao",
      "url": "https://www.jinhenghaoblog.top"
    }
  ]
}</script><link rel="shortcut icon" href="/images/favicon.ico"><link rel="canonical" href="https://www.jinhenghaoblog.top/posts/20250903205237.html"><link rel="preconnect" href="//cdn.jsdelivr.net"><link rel="preconnect" href="//busuanzi.ibruce.info"><meta name="google-site-verification" content="YCuKDW04QOezpE8jWTLZ_yiyayYEZjHGT3P_qo9C13I"><meta name="baidu-site-verification" content="codeva-Wv2xaBiiDf"><link rel="stylesheet" href="/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free/css/all.min.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/node-snackbar/dist/snackbar.min.css" media="print" onload='this.media="all"'><script>(()=>{var e={set:(e,t,a)=>{a&&(a=Date.now()+864e5*a,localStorage.setItem(e,JSON.stringify({value:t,expiry:a})))},get:e=>{var t=localStorage.getItem(e);if(t){var{value:t,expiry:a}=JSON.parse(t);if(!(Date.now()>a))return t;localStorage.removeItem(e)}}},t=(window.btf={saveToLocal:e,getScript:(o,n={})=>new Promise((e,t)=>{let a=document.createElement("script");a.src=o,a.async=!0,Object.entries(n).forEach(([e,t])=>a.setAttribute(e,t)),a.onload=a.onreadystatechange=()=>{a.readyState&&!/loaded|complete/.test(a.readyState)||e()},a.onerror=t,document.head.appendChild(a)}),getCSS:(o,n)=>new Promise((e,t)=>{let a=document.createElement("link");a.rel="stylesheet",a.href=o,n&&(a.id=n),a.onload=a.onreadystatechange=()=>{a.readyState&&!/loaded|complete/.test(a.readyState)||e()},a.onerror=t,document.head.appendChild(a)}),addGlobalFn:(e,t,a=!1,o=window)=>{var n;e.startsWith("pjax")||((n=o.globalFn||{})[e]=n[e]||{},n[e][a||Object.keys(n[e]).length]=t,o.globalFn=n)}},()=>{document.documentElement.setAttribute("data-theme","dark"),null!==document.querySelector('meta[name="theme-color"]')&&document.querySelector('meta[name="theme-color"]').setAttribute("content","#0d0d0d")}),a=()=>{document.documentElement.setAttribute("data-theme","light"),null!==document.querySelector('meta[name="theme-color"]')&&document.querySelector('meta[name="theme-color"]').setAttribute("content","#ffffff")},o=(btf.activateDarkMode=t,btf.activateLightMode=a,e.get("theme")),t=("dark"===o?t():"light"===o&&a(),e.get("aside-status"));void 0!==t&&document.documentElement.classList.toggle("hide-aside","hide"===t);/iPad|iPhone|iPod|Macintosh/.test(navigator.userAgent)&&document.documentElement.classList.add("apple")})()</script><script>let GLOBAL_CONFIG={root:"/",algolia:void 0,localSearch:void 0,translate:void 0,highlight:{plugin:"highlight.js",highlightCopy:!0,highlightLang:!0,highlightHeightLimit:!1,highlightFullpage:!1,highlightMacStyle:!1},copy:{success:"复制成功",error:"复制失败",noSupport:"浏览器不支持"},relativeDate:{homepage:!1,post:!1},runtime:"",dateSuffix:{just:"刚刚",min:"分钟前",hour:"小时前",day:"天前",month:"个月前"},copyright:void 0,lightbox:"null",Snackbar:{chs_to_cht:"已切换为繁体中文",cht_to_chs:"已切换为简体中文",day_to_night:"已切换为深色模式",night_to_day:"已切换为浅色模式",bgLight:"#49b1f5",bgDark:"#1f1f1f",position:"bottom-left"},infinitegrid:{js:"https://cdn.jsdelivr.net/npm/@egjs/infinitegrid/dist/infinitegrid.min.js",buttonText:"加载更多"},isPhotoFigcaption:!1,islazyloadPlugin:!1,isAnchor:!1,percent:{toc:!0,rightside:!1},autoDarkmode:!1}</script><script id="config-diff">var GLOBAL_CONFIG_SITE={title:"RNA-seq基本介绍及实战——差异分析",isHighlightShrink:!1,isToc:!0,pageType:"post"}</script><meta name="generator" content="Hexo 7.3.0"><link rel="alternate" href="/atom.xml" title="JinhengHao的博客" type="application/atom+xml"></head><body><div id="sidebar"><div id="menu-mask"></div><div id="sidebar-menus"><div class="avatar-img text-center"><img src="/img/butterfly-icon.jpg" onerror='this.onerror=null,this.src="/img/friend_404.gif"' alt="avatar"></div><div class="site-data text-center"><a href="/archives/"><div class="headline">文章</div><div class="length-num">23</div></a><a href="/tags/"><div class="headline">标签</div><div class="length-num">14</div></a><a href="/categories/"><div class="headline">分类</div><div class="length-num">7</div></a></div><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fas fa-home"></i><span> 主页</span></a></div><div class="menus_item"><a class="site-page" href="/search/"><i class="fa-fw fas fa-folder-open"></i><span> 搜索</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fas fa-folder-open"></i><span> 分类</span></a></div><div class="menus_item"><a class="site-page" href="/about/"><i class="fa-fw fas fa-folder-open"></i><span> 关于</span></a></div></div></div></div><div class="post" id="body-wrap"><header class="post-bg" id="page-header" style="background-image:url(/img/background.png)"><nav id="nav"><span id="blog-info"><a class="nav-site-title" href="/"><img class="site-icon" src="/img/butterfly-icon.jpg" alt="Logo"><span class="site-name">JinhengHao的博客</span></a><a class="nav-page-title" href="/"><span class="site-name">RNA-seq基本介绍及实战——差异分析</span><span class="site-name"><i class="fa-solid fa-circle-arrow-left"></i><span> 返回首页</span></span></a></span><div id="menus"><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fas fa-home"></i><span> 主页</span></a></div><div class="menus_item"><a class="site-page" href="/search/"><i class="fa-fw fas fa-folder-open"></i><span> 搜索</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fas fa-folder-open"></i><span> 分类</span></a></div><div class="menus_item"><a class="site-page" href="/about/"><i class="fa-fw fas fa-folder-open"></i><span> 关于</span></a></div></div><div id="toggle-menu"><span class="site-page"><i class="fas fa-bars fa-fw"></i></span></div></div></nav><div id="post-info"><h1 class="post-title">RNA-seq基本介绍及实战——差异分析</h1><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">发表于</span><time class="post-meta-date-created" datetime="2025-09-03T12:52:37.000Z" title="发表于 2025-09-03 20:52:37">2025-09-03</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">更新于</span><time class="post-meta-date-updated" datetime="2025-09-12T09:47:50.146Z" title="更新于 2025-09-12 17:47:50">2025-09-12</time></span><span class="post-meta-categories"><span class="post-meta-separator">|</span><i class="fas fa-inbox fa-fw post-meta-icon"></i><a class="post-meta-categories" href="/categories/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF/">生物信息</a><i class="fas fa-angle-right post-meta-separator"></i><i class="fas fa-inbox fa-fw post-meta-icon"></i><a class="post-meta-categories" href="/categories/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF/%E5%AD%A6%E4%B9%A0/">学习</a></span></div><div class="meta-secondline"><span class="post-meta-separator">|</span><span data-flag-title=""><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">浏览量:</span><span id="twikoo_visitors"><i class="fa-solid fa-spinner fa-spin"></i></span></span><span class="post-meta-separator">|</span><span class="post-meta-commentcount"><i class="far fa-comments fa-fw post-meta-icon"></i><span class="post-meta-label">评论数:</span><a href="/posts/20250903205237.html#post-comment"><span id="twikoo-count"><i class="fa-solid fa-spinner fa-spin"></i></span></a></span></div></div></div></header><main class="layout" id="content-inner"><div id="post"><article class="container post-content" id="article-container"><h2 id="实战环节"><a href="#实战环节" class="headerlink" title="实战环节"></a>实战环节</h2><p>上一期对所拥有的数据做了个表达定量并生成了表达矩阵，现在进入差异分析的环节</p><h2 id="差异分析"><a href="#差异分析" class="headerlink" title="差异分析"></a>差异分析</h2><p>本次做差异分析所使用的工具是基于python的omicverse库</p><p>该模块的其安装方法也很简单，但需注意的是omicverse库必须在linux环境或windows系统的WSL环境下使用。</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line"><span class="comment"># 使用conda</span></span><br><span class="line">conda create -n omicverse python=3.10</span><br><span class="line">conda activate omicverse</span><br><span class="line"></span><br><span class="line"><span class="comment"># 安装pytorch-gpu版 （二选一）</span></span><br><span class="line">conda install pytorch torchvision torchaudio pytorch-cuda=11.8 -c pytorch -c nvidia</span><br><span class="line"></span><br><span class="line"><span class="comment"># 安装putorch-cpu版 （二选一）</span></span><br><span class="line">conda install pytorch torchvision torchaudio cpuonly -c pytorch</span><br><span class="line"></span><br><span class="line"><span class="comment"># 安装pyg</span></span><br><span class="line">conda install pyg -c pyg</span><br><span class="line"></span><br><span class="line"><span class="comment"># 安装omicverse</span></span><br><span class="line">conda install omicverse -c conda-forge</span><br></pre></td></tr></table></figure><p>本次分析选用的数据集是GSE118916。从GEO数据库中进行下载：</p><h2 id="使用omicverse进行常规的差异基因表达分析"><a href="#使用omicverse进行常规的差异基因表达分析" class="headerlink" title="使用omicverse进行常规的差异基因表达分析"></a>使用omicverse进行常规的差异基因表达分析</h2><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">import</span> pandas <span class="keyword">as</span> pd</span><br><span class="line"><span class="keyword">import</span> os</span><br></pre></td></tr></table></figure><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 数据预处理</span></span><br><span class="line">expressionData = pd.read_csv(<span class="string">&quot;expression_data.csv&quot;</span>, encoding=<span class="string">&#x27;UTF-8&#x27;</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 将没有基因名的数据索引拿出来</span></span><br><span class="line">indexNum = <span class="number">0</span></span><br><span class="line">indexList = []</span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> <span class="built_in">list</span>(expressionData.iloc[:,<span class="number">0</span>]):</span><br><span class="line">    <span class="keyword">if</span> <span class="string">&#x27;---&#x27;</span> == i:</span><br><span class="line">        <span class="comment"># print(indexNum)</span></span><br><span class="line">        indexList.append(indexNum)</span><br><span class="line">    indexNum += <span class="number">1</span></span><br><span class="line"></span><br><span class="line">expressionData_Drop = expressionData.drop(indexList)</span><br><span class="line">expressionData_Drop</span><br></pre></td></tr></table></figure><p>输出：</p><table><thead><tr><th></th><th>Gene Symbol</th><th>GSM3351221</th><th>GSM3351222</th><th>GSM3351223</th><th>GSM3351224</th><th>GSM3351225</th><th>GSM3351226</th><th>GSM3351227</th><th>GSM3351228</th><th>…</th><th>GSM3351240</th><th>GSM3351241</th><th>GSM3351242</th><th>GSM3351243</th><th>GSM3351244</th><th>GSM3351245</th><th>GSM3351246</th><th>GSM3351247</th><th>GSM3351248</th><th>GSM3351249</th></tr></thead><tbody><tr><td>0</td><td>HIST1H3G</td><td>4.333162</td><td>3.393413</td><td>3.202298</td><td>3.273934</td><td>3.149646</td><td>3.371994</td><td>3.627498</td><td>3.046871</td><td>3.969594</td><td>…</td><td>3.623850</td><td>3.664705</td><td>3.340904</td><td>3.602020</td><td>3.331657</td><td>3.392765</td><td>3.493841</td><td>3.487211</td><td>3.391482</td></tr><tr><td>1</td><td>HIST1H3G</td><td>5.509134</td><td>5.405715</td><td>5.184302</td><td>4.883117</td><td>4.896176</td><td>5.113827</td><td>5.167449</td><td>5.110204</td><td>5.556905</td><td>…</td><td>4.851287</td><td>5.397218</td><td>5.153698</td><td>5.201280</td><td>4.851287</td><td>5.374734</td><td>5.276653</td><td>5.188680</td><td>5.140900</td></tr><tr><td>2</td><td>HIST1H3G</td><td>4.686904</td><td>3.871414</td><td>3.724224</td><td>3.397720</td><td>3.615277</td><td>3.841320</td><td>4.325271</td><td>3.560210</td><td>4.341479</td><td>…</td><td>3.992363</td><td>3.987619</td><td>3.687451</td><td>3.855043</td><td>3.694976</td><td>3.766741</td><td>3.963058</td><td>4.101157</td><td>3.854747</td></tr><tr><td>3</td><td>TNFAIP8L1</td><td>4.361757</td><td>4.285916</td><td>4.377757</td><td>5.367287</td><td>6.018805</td><td>4.888754</td><td>4.743503</td><td>4.430155</td><td>4.498028</td><td>…</td><td>3.740284</td><td>3.791165</td><td>4.206512</td><td>3.795124</td><td>3.853014</td><td>3.782595</td><td>3.757001</td><td>3.663455</td><td>3.869301</td></tr><tr><td>4</td><td>OTOP2</td><td>4.692442</td><td>4.467624</td><td>4.430727</td><td>4.132381</td><td>4.265175</td><td>4.237638</td><td>4.287581</td><td>4.428147</td><td>4.352447</td><td>…</td><td>4.834237</td><td>4.873393</td><td>4.521720</td><td>4.899339</td><td>5.179813</td><td>4.866155</td><td>5.235878</td><td>4.829041</td><td>4.766776</td></tr><tr><td>…</td><td>…</td><td>…</td><td>…</td><td>…</td><td>…</td><td>…</td><td>…</td><td>…</td><td>…</td><td>…</td><td>…</td><td>…</td><td>…</td><td>…</td><td>…</td><td>…</td><td>…</td><td>…</td><td>…</td><td>…</td></tr><tr><td>49297</td><td>GAPDH</td><td>2.250415</td><td>2.196336</td><td>2.188453</td><td>2.140965</td><td>2.205614</td><td>2.172010</td><td>2.216602</td><td>2.180626</td><td>2.197528</td><td>…</td><td>2.311198</td><td>2.319124</td><td>2.243291</td><td>2.354154</td><td>2.397206</td><td>2.348041</td><td>2.317698</td><td>2.321311</td><td>2.272872</td></tr><tr><td>49298</td><td>STAT1</td><td>2.349588</td><td>2.322962</td><td>2.277523</td><td>2.217424</td><td>2.293500</td><td>2.242945</td><td>2.283375</td><td>2.278833</td><td>2.299161</td><td>…</td><td>2.439090</td><td>2.465033</td><td>2.348560</td><td>2.472626</td><td>2.507350</td><td>2.452186</td><td>2.461888</td><td>2.474203</td><td>2.423064</td></tr><tr><td>49299</td><td>STAT1</td><td>2.671896</td><td>2.611722</td><td>2.612591</td><td>2.468304</td><td>2.585089</td><td>2.535129</td><td>2.606042</td><td>2.542626</td><td>2.584898</td><td>…</td><td>2.811227</td><td>2.805426</td><td>2.675989</td><td>2.881790</td><td>2.842771</td><td>2.828554</td><td>2.840772</td><td>2.815630</td><td>2.776199</td></tr><tr><td>49300</td><td>STAT1</td><td>3.041021</td><td>3.000218</td><td>2.956512</td><td>2.836378</td><td>2.925315</td><td>2.865108</td><td>2.960598</td><td>2.930377</td><td>2.947940</td><td>…</td><td>3.221686</td><td>3.264126</td><td>3.072335</td><td>3.282286</td><td>3.309083</td><td>3.238734</td><td>3.269109</td><td>3.202032</td><td>3.186877</td></tr><tr><td>49301</td><td>STAT1</td><td>3.521169</td><td>3.519029</td><td>3.485920</td><td>3.299582</td><td>3.402691</td><td>3.347042</td><td>3.427732</td><td>3.405621</td><td>3.408984</td><td>…</td><td>3.757983</td><td>3.837290</td><td>3.611317</td><td>3.841599</td><td>3.827958</td><td>3.830761</td><td>3.752328</td><td>3.749842</td><td>3.681698</td></tr></tbody></table><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 将第一列设置为表格的index</span></span><br><span class="line">expressionData_Drop.set_index([<span class="string">&quot;Gene Symbol&quot;</span>], inplace=<span class="literal">True</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 导入omicverse库</span></span><br><span class="line"><span class="keyword">import</span> omicverse <span class="keyword">as</span> ov</span><br><span class="line"><span class="keyword">import</span> scanpy <span class="keyword">as</span> sc</span><br><span class="line"><span class="keyword">import</span> matplotlib.pyplot <span class="keyword">as</span> plt</span><br><span class="line">ov.ov_plot_set()</span><br></pre></td></tr></table></figure><p>输出：</p><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">🔬 Starting plot initialization...</span><br><span class="line">🧬 Detecting CUDA devices…</span><br><span class="line">🚫 No CUDA devices found</span><br><span class="line">✅ plot_set complete.</span><br></pre></td></tr></table></figure><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 差异分析</span></span><br><span class="line">dds = ov.bulk.pyDEG(expressionData_Drop)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 该步骤是用于清洗数据，将重复的基因去除，保留表达量高的基因</span></span><br><span class="line">dds.drop_duplicates_index()</span><br><span class="line"><span class="built_in">print</span>(<span class="string">&#x27;... drop_duplicates_index success&#x27;</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 数据标准化，去除表中可能的批次效应</span></span><br><span class="line">dds.normalize()</span><br><span class="line"><span class="built_in">print</span>(<span class="string">&#x27;... estimateSizeFactors and normalize success&#x27;</span>)</span><br></pre></td></tr></table></figure><p>输出：</p><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">... </span>drop_duplicates_index success</span><br><span class="line"><span class="meta">... </span>estimateSizeFactors <span class="keyword">and</span> normalize success</span><br></pre></td></tr></table></figure><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 从使用t检验方法计算矩阵中的差异表达基因，需要输入处理组和对照组。</span></span><br><span class="line"><span class="comment"># 处理组</span></span><br><span class="line">treatment_groups=[<span class="string">&#x27;GSM3351220&#x27;</span>,<span class="string">&#x27;GSM3351221&#x27;</span>,<span class="string">&#x27;GSM3351222&#x27;</span>,<span class="string">&#x27;GSM3351223&#x27;</span>,<span class="string">&#x27;GSM3351224&#x27;</span>,<span class="string">&#x27;GSM3351225&#x27;</span>,<span class="string">&#x27;GSM3351226&#x27;</span>,<span class="string">&#x27;GSM3351227&#x27;</span>,<span class="string">&#x27;GSM3351228&#x27;</span>,<span class="string">&#x27;GSM3351229&#x27;</span>,<span class="string">&#x27;GSM3351230&#x27;</span>,<span class="string">&#x27;GSM3351231&#x27;</span>,<span class="string">&#x27;GSM3351232&#x27;</span>,<span class="string">&#x27;GSM3351233&#x27;</span>,<span class="string">&#x27;GSM3351234&#x27;</span>]</span><br><span class="line"><span class="comment"># 对照组</span></span><br><span class="line">control_groups=[<span class="string">&#x27;GSM3351235&#x27;</span>,<span class="string">&#x27;GSM3351236&#x27;</span>,<span class="string">&#x27;GSM3351237&#x27;</span>,<span class="string">&#x27;GSM3351238&#x27;</span>,<span class="string">&#x27;GSM3351239&#x27;</span>,<span class="string">&#x27;GSM3351240&#x27;</span>,<span class="string">&#x27;GSM3351241&#x27;</span>,<span class="string">&#x27;GSM3351242&#x27;</span>,<span class="string">&#x27;GSM3351243&#x27;</span>,<span class="string">&#x27;GSM3351244&#x27;</span>,<span class="string">&#x27;GSM3351245&#x27;</span>,<span class="string">&#x27;GSM3351246&#x27;</span>,<span class="string">&#x27;GSM3351247&#x27;</span>,<span class="string">&#x27;GSM3351248&#x27;</span>,<span class="string">&#x27;GSM3351249&#x27;</span>]</span><br><span class="line">result_ttest=dds.deg_analysis(treatment_groups,control_groups,method=<span class="string">&#x27;ttest&#x27;</span>)</span><br><span class="line">result_ttest.sort_values(<span class="string">&#x27;qvalue&#x27;</span>).head()</span><br></pre></td></tr></table></figure><p>输出：</p><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">⚙️ You are using ttest method for differential expression analysis.</span><br><span class="line">⏰ Start to calculate qvalue...</span><br><span class="line">✅ Differential expression analysis completed.</span><br></pre></td></tr></table></figure><table><thead><tr><th>Gene Symbol</th><th>pvalue</th><th>qvalue</th><th>FoldChange</th><th>MaxBaseMean</th><th>BaseMean</th><th>log2(BaseMean)</th><th>log2FC</th><th>abs(log2FC)</th><th>size</th><th>-log(pvalue)</th><th>-log(qvalue)</th><th>sig</th></tr></thead><tbody><tr><td>DMRTA1</td><td>3.372637e-17</td><td>6.773267e-13</td><td>0.715917</td><td>5.498099</td><td>4.453653</td><td>2.154989</td><td>-0.482136</td><td>0.482136</td><td>0.071592</td><td>16.472030</td><td>12.169202</td><td>sig</td></tr><tr><td>CAPN13</td><td>1.526491e-15</td><td>1.532826e-11</td><td>0.849719</td><td>6.694391</td><td>6.051985</td><td>2.597408</td><td>-0.234943</td><td>0.234943</td><td>0.084972</td><td>14.816306</td><td>10.814507</td><td>sig</td></tr><tr><td>LNX1</td><td>3.725194e-15</td><td>2.493769e-11</td><td>0.763073</td><td>6.469154</td><td>5.483045</td><td>2.454977</td><td>-0.390108</td><td>0.390108</td><td>0.076307</td><td>14.428851</td><td>10.603144</td><td>sig</td></tr><tr><td>CAPN9</td><td>5.643624e-15</td><td>2.833522e-11</td><td>0.667939</td><td>7.792100</td><td>6.190386</td><td>2.630029</td><td>-0.582212</td><td>0.582212</td><td>0.066794</td><td>14.248442</td><td>10.547673</td><td>sig</td></tr><tr><td>CLDN12</td><td>7.299049e-15</td><td>2.931736e-11</td><td>0.903028</td><td>6.111705</td><td>5.725430</td><td>2.517384</td><td>-0.147158</td><td>0.147158</td><td>0.090303</td><td>14.136734</td><td>10.532875</td><td>sig</td></tr></tbody></table><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 一个重要事项是，在处理差异表达基因（DEGs）时并未过滤低表达基因，在后续版本中会加入相应的处理功能。</span></span><br><span class="line"><span class="built_in">print</span>(dds.result.shape)</span><br><span class="line">dds.result=dds.result.loc[dds.result[<span class="string">&#x27;log2(BaseMean)&#x27;</span>]&gt;<span class="number">1</span>]</span><br><span class="line"><span class="built_in">print</span>(dds.result.shape)</span><br><span class="line"></span><br><span class="line">输出：</span><br><span class="line">(<span class="number">14028</span>, <span class="number">12</span>)</span><br><span class="line">(<span class="number">14028</span>, <span class="number">12</span>)</span><br></pre></td></tr></table></figure><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 下一步还需要设置Foldchange（倍数变化）的阈值，需要准备一个名为foldchange_set的方法来完成这一任务。</span></span><br><span class="line"><span class="comment"># 该功能会根据log2FC（以2为底的对数倍数变化）的分布自动计算合适的阈值，但也可以手动输入。</span></span><br><span class="line"><span class="comment"># -1 means automatically calculates</span></span><br><span class="line">dds.foldchange_set(fc_threshold=-<span class="number">1</span>,</span><br><span class="line">                   pval_threshold=<span class="number">0.05</span>,</span><br><span class="line">                   logp_max=<span class="number">10</span>)</span><br></pre></td></tr></table></figure><p>输出：</p><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">... Fold change threshold: 0.16936119706053465</span><br></pre></td></tr></table></figure><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 可视化差异表达基因（DEG）结果与特定基因</span></span><br><span class="line"><span class="comment"># 为可视化DEG分析结果，使用plot_volcano函数进行绘制。该函数可重点展示目标基因或高差异表达基因，需要输入以下参数：</span></span><br><span class="line"><span class="comment"># title：火山图标题</span></span><br><span class="line"><span class="comment"># figsize：图像尺寸</span></span><br><span class="line"><span class="comment"># plot_genes：需重点标注的目标基因列表</span></span><br><span class="line"><span class="comment"># plot_genes_num：若无特定目标基因，可设置自动标注的基因数量</span></span><br><span class="line">dds.plot_volcano(title=<span class="string">&#x27;DEG Analysis&#x27;</span>,figsize=(<span class="number">10</span>,<span class="number">10</span>),</span><br><span class="line">                 plot_genes_num=<span class="number">0</span>,plot_genes_fontsize=<span class="number">0</span>,)</span><br></pre></td></tr></table></figure><p>输出：</p><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line">🌋 Volcano Plot Analysis:</span><br><span class="line">   Total genes: 14028</span><br><span class="line">   ↗️  Upregulated genes: 798</span><br><span class="line">   ↘️  Downregulated genes: 643</span><br><span class="line">   ➡️  Non-significant genes: 12587</span><br><span class="line">   🎯 Total significant genes: 1441</span><br><span class="line">   log2FC range: -0.99 to 0.79</span><br><span class="line">   qvalue range: 6.77e-13 to 1.00e+00</span><br><span class="line"></span><br><span class="line">⚙️  Current Function Parameters:</span><br><span class="line">   Data columns: pval_name=&#x27;qvalue&#x27;, fc_name=&#x27;log2FC&#x27;</span><br><span class="line">   Thresholds: pval_threshold=0.05, fc_max=0.16936119706053465, fc_min=-0.16936119706053465</span><br><span class="line">   Plot size: figsize=(10, 10)</span><br><span class="line">   Gene labels: plot_genes_num=0, plot_genes_fontsize=0</span><br><span class="line">   Custom genes: None (auto-select top genes)</span><br><span class="line"></span><br><span class="line">💡 Parameter Optimization Suggestions:</span><br><span class="line">   ✅ Current parameters are optimal for your data!</span><br><span class="line">────────────────────────────────────────────────────────────</span><br></pre></td></tr></table></figure><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">&lt;AxesSubplot: title=&#123;&#x27;center&#x27;: &#x27;DEG Analysis&#x27;&#125;, xlabel=&#x27;$log_&#123;2&#125;FC$&#x27;, ylabel=&#x27;$-log_&#123;10&#125;(qvalue)$&#x27;&gt;</span><br></pre></td></tr></table></figure><p><img src="/images/image_1757669142014.png" alt="image.png"></p><h2 id="使用-DEseq2-进行差异表达分析"><a href="#使用-DEseq2-进行差异表达分析" class="headerlink" title="使用 DEseq2 进行差异表达分析"></a>使用 DEseq2 进行差异表达分析</h2><p>借助 pyDEseq2，可以像 R 语言那样分析差异表达基因</p><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 将表达矩阵全部转换为整数</span></span><br><span class="line">expressionData_int = expressionData_Drop.astype(<span class="built_in">int</span>)</span><br><span class="line"><span class="comment"># expressionData_int.head()</span></span><br><span class="line"></span><br><span class="line">dds_int = ov.bulk.pyDEG(expressionData_int)</span><br><span class="line"></span><br><span class="line">dds_int.drop_duplicates_index()</span><br><span class="line"><span class="built_in">print</span>(<span class="string">&#x27;... drop_duplicates_index success&#x27;</span>)</span><br></pre></td></tr></table></figure><p>输出：</p><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">... drop_duplicates_index success</span><br></pre></td></tr></table></figure><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 从使用DEseq2方法计算矩阵中的差异表达基因，需要输入处理组和对照组。</span></span><br><span class="line"><span class="comment"># 处理组</span></span><br><span class="line">treatment_groups=[<span class="string">&#x27;GSM3351220&#x27;</span>,<span class="string">&#x27;GSM3351221&#x27;</span>,<span class="string">&#x27;GSM3351222&#x27;</span>,<span class="string">&#x27;GSM3351223&#x27;</span>,<span class="string">&#x27;GSM3351224&#x27;</span>,<span class="string">&#x27;GSM3351225&#x27;</span>,<span class="string">&#x27;GSM3351226&#x27;</span>,<span class="string">&#x27;GSM3351227&#x27;</span>,<span class="string">&#x27;GSM3351228&#x27;</span>,<span class="string">&#x27;GSM3351229&#x27;</span>,<span class="string">&#x27;GSM3351230&#x27;</span>,<span class="string">&#x27;GSM3351231&#x27;</span>,<span class="string">&#x27;GSM3351232&#x27;</span>,<span class="string">&#x27;GSM3351233&#x27;</span>,<span class="string">&#x27;GSM3351234&#x27;</span>]</span><br><span class="line"><span class="comment"># 对照组</span></span><br><span class="line">control_groups=[<span class="string">&#x27;GSM3351235&#x27;</span>,<span class="string">&#x27;GSM3351236&#x27;</span>,<span class="string">&#x27;GSM3351237&#x27;</span>,<span class="string">&#x27;GSM3351238&#x27;</span>,<span class="string">&#x27;GSM3351239&#x27;</span>,<span class="string">&#x27;GSM3351240&#x27;</span>,<span class="string">&#x27;GSM3351241&#x27;</span>,<span class="string">&#x27;GSM3351242&#x27;</span>,<span class="string">&#x27;GSM3351243&#x27;</span>,<span class="string">&#x27;GSM3351244&#x27;</span>,<span class="string">&#x27;GSM3351245&#x27;</span>,<span class="string">&#x27;GSM3351246&#x27;</span>,<span class="string">&#x27;GSM3351247&#x27;</span>,<span class="string">&#x27;GSM3351248&#x27;</span>,<span class="string">&#x27;GSM3351249&#x27;</span>]</span><br><span class="line">result = dds_int.deg_analysis(treatment_groups,control_groups,method=<span class="string">&#x27;DEseq2&#x27;</span>)</span><br><span class="line"><span class="comment"># result_ttest.sort_values(&#x27;qvalue&#x27;).head()</span></span><br></pre></td></tr></table></figure><p>输出：</p><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">⚙️ You are using DEseq2 method for differential expression analysis.</span><br><span class="line">⏰ Start to create DeseqDataSet...</span><br><span class="line">Using None as control genes, passed at DeseqDataSet initialization</span><br></pre></td></tr></table></figure><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">Fitting size factors...</span><br><span class="line">... done in 0.05 seconds.</span><br><span class="line"></span><br><span class="line">Fitting dispersions...</span><br><span class="line">... done in 0.65 seconds.</span><br><span class="line"></span><br><span class="line">Fitting dispersion trend curve...</span><br><span class="line">... done in 0.22 seconds.</span><br><span class="line"></span><br><span class="line">Fitting MAP dispersions...</span><br></pre></td></tr></table></figure><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">logres_prior=1.5254760032591015, sigma_prior=1.4514357345950912</span><br></pre></td></tr></table></figure><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">... done in 1.82 seconds.</span><br><span class="line"></span><br><span class="line">Fitting LFCs...</span><br><span class="line">... done in 0.77 seconds.</span><br><span class="line"></span><br><span class="line">Calculating cook&#x27;s distance...</span><br><span class="line">... done in 0.02 seconds.</span><br><span class="line"></span><br><span class="line">Replacing 0 outlier genes.</span><br><span class="line"></span><br><span class="line">Running Wald tests...</span><br></pre></td></tr></table></figure><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br></pre></td><td class="code"><pre><span class="line">Log2 fold change &amp; Wald test p-value: condition Treatment vs Control</span><br><span class="line">                              baseMean  log2FoldChange     lfcSE      stat  \</span><br><span class="line">Gene Symbol                                                                </span><br><span class="line">COX3 /// MYBPC3 /// SLC12A1  13.074116        0.001168  0.154571  0.007559   </span><br><span class="line">TPT1                         13.007369        0.001171  0.154922  0.007558   </span><br><span class="line">ND4                          13.007369        0.001171  0.154922  0.007558   </span><br><span class="line">ATP13A5 /// HARS             12.974036       -0.006246  0.155129 -0.040261   </span><br><span class="line">B2M                          12.840421       -0.021317  0.155953 -0.136689   </span><br><span class="line">...                                ...             ...       ...       ...   </span><br><span class="line">SSX1                          1.300795       -0.072834  0.473870 -0.153699   </span><br><span class="line">C2orf83                       1.300624       -0.221223  0.475017 -0.465715   </span><br><span class="line">CLCA3P                        1.233980       -0.551375  0.494168 -1.115764   </span><br><span class="line">UGT2B11                       1.200886       -0.484293  0.501002 -0.966649   </span><br><span class="line">RAD21L1                       1.233900       -0.076840  0.486072 -0.158084   </span><br><span class="line"></span><br><span class="line">                               pvalue      padj  </span><br><span class="line">Gene Symbol                                    </span><br><span class="line">COX3 /// MYBPC3 /// SLC12A1  0.993969  0.997498  </span><br><span class="line">TPT1                         0.993969  0.997498  </span><br><span class="line">ND4                          0.993969  0.997498  </span><br><span class="line">ATP13A5 /// HARS             0.967885  0.997498  </span><br><span class="line">B2M                          0.891277  0.997498  </span><br><span class="line">...                               ...       ...  </span><br><span class="line">SSX1                         0.877847       NaN  </span><br><span class="line">C2orf83                      0.641419       NaN  </span><br><span class="line">CLCA3P                       0.264523       NaN  </span><br><span class="line">UGT2B11                      0.333719       NaN  </span><br><span class="line">RAD21L1                      0.874391       NaN  </span><br><span class="line"></span><br><span class="line">[20083 rows x 6 columns]</span><br><span class="line">✅ Differential expression analysis completed.</span><br></pre></td></tr></table></figure><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">... done in 1.17 seconds.</span><br></pre></td></tr></table></figure><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">print</span>(result.shape)</span><br><span class="line">result=result.loc[result[<span class="string">&#x27;log2(BaseMean)&#x27;</span>]&gt;<span class="number">1</span>]</span><br><span class="line"><span class="built_in">print</span>(result.shape)</span><br></pre></td></tr></table></figure><p>输出：</p><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">(20083, 14)</span><br><span class="line">(20083, 14)</span><br></pre></td></tr></table></figure><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 需要设置Foldchange（倍数变化）的阈值</span></span><br><span class="line">dds_int.foldchange_set(fc_threshold=-<span class="number">1</span>,</span><br><span class="line">                   pval_threshold=<span class="number">0.05</span>,</span><br><span class="line">                   logp_max=<span class="number">10</span>)</span><br></pre></td></tr></table></figure><p>输出：</p><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">... Fold change threshold: 0.2874723264386634</span><br></pre></td></tr></table></figure><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 差异基因的可视化</span></span><br><span class="line">dds.plot_volcano(title=<span class="string">&#x27;DEG Analysis&#x27;</span>,figsize=(<span class="number">10</span>,<span class="number">10</span>),</span><br><span class="line">                 plot_genes_num=<span class="number">8</span>,plot_genes_fontsize=<span class="number">12</span>,)</span><br></pre></td></tr></table></figure><p>输出：</p><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line">🌋 Volcano Plot Analysis:</span><br><span class="line">   Total genes: 14028</span><br><span class="line">   ↗️  Upregulated genes: 798</span><br><span class="line">   ↘️  Downregulated genes: 643</span><br><span class="line">   ➡️  Non-significant genes: 12587</span><br><span class="line">   🎯 Total significant genes: 1441</span><br><span class="line">   log2FC range: -0.99 to 0.79</span><br><span class="line">   qvalue range: 6.77e-13 to 1.00e+00</span><br><span class="line"></span><br><span class="line">⚙️  Current Function Parameters:</span><br><span class="line">   Data columns: pval_name=&#x27;qvalue&#x27;, fc_name=&#x27;log2FC&#x27;</span><br><span class="line">   Thresholds: pval_threshold=0.05, fc_max=0.16936119706053465, fc_min=-0.16936119706053465</span><br><span class="line">   Plot size: figsize=(10, 10)</span><br><span class="line">   Gene labels: plot_genes_num=8, plot_genes_fontsize=12</span><br><span class="line">   Custom genes: None (auto-select top genes)</span><br><span class="line"></span><br><span class="line">💡 Parameter Optimization Suggestions:</span><br><span class="line">   ✅ Current parameters are optimal for your data!</span><br><span class="line">────────────────────────────────────────────────────────────</span><br></pre></td></tr></table></figure><p><img src="/images/image_1757670416755.png" alt="image.png"></p><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 特定基因表达量分析</span></span><br><span class="line">dds_int.plot_boxplot(genes=[<span class="string">&#x27;CAPN9&#x27;</span>,<span class="string">&#x27;LNX1&#x27;</span>,<span class="string">&#x27;PFDN2&#x27;</span>,<span class="string">&#x27;CDK4&#x27;</span>],treatment_groups=treatment_groups,</span><br><span class="line">                control_groups=control_groups,figsize=(<span class="number">4</span>,<span class="number">6</span>),fontsize=<span class="number">12</span>,</span><br><span class="line">                 legend_bbox=(<span class="number">1</span>,<span class="number">0.55</span>))</span><br></pre></td></tr></table></figure><p>输出：<br><img src="/images/image_1757670446701.png" alt="image.png"></p><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 单个基因的差异分析</span></span><br><span class="line">dds_int.plot_boxplot(genes=[<span class="string">&#x27;CAPN9&#x27;</span>],treatment_groups=treatment_groups,</span><br><span class="line">                control_groups=control_groups,figsize=(<span class="number">2</span>,<span class="number">3</span>),fontsize=<span class="number">12</span>,</span><br><span class="line">                 legend_bbox=(<span class="number">2</span>,<span class="number">0.55</span>))</span><br></pre></td></tr></table></figure><p>输出：</p><p><img src="/images/image_1757670467621.png" alt="image.png"></p></article><div class="tag_share"><div class="post-meta__tag-list"><a class="post-meta__tags" href="/tags/R/">R</a><a class="post-meta__tags" href="/tags/python/">python</a><a class="post-meta__tags" href="/tags/%E8%BD%AC%E5%BD%95%E7%BB%84%E5%AD%A6/">转录组学</a></div><div class="post-share"><div class="social-share" data-image="/img/butterfly-icon.jpg" data-sites="wechat,weibo,qq"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/butterfly-extsrc/sharejs/dist/css/share.min.css" media="print" onload='this.media="all"'><script src="https://cdn.jsdelivr.net/npm/butterfly-extsrc/sharejs/dist/js/social-share.min.js" defer></script></div></div><nav class="pagination-post" id="pagination"><a class="pagination-related" href="/posts/20250916200356.html" title="基于David数据库与微生信平台进行Go-KEGG分析"><div class="cover" style="background:var(--default-bg-color)"></div><div class="info"><div class="info-1"><div class="info-item-1">上一篇</div><div class="info-item-2">基于David数据库与微生信平台进行Go-KEGG分析</div></div><div class="info-2"><div class="info-item-1">进入DAVID数据库 选择Functional Annotation 点击Functional Annotation Tool 选择GO与PathWays GO这里选择BP、CC、MF 点击Chart可看到： 点击下载文件，将数据复制粘贴到EXCEL中备用。 其余的CC、MF也是同样的操作 PATHWAY这里我们只选择KEGG进行下载，也是点击Chart，下载文件并整理到excel中备用。 整理从DAVID数据库下载的文件一般下载下来的表格如图所示： 在这里分了4个Sheet，每个Sheet对应着从不同地方搜集下来的数据，分为BP、CC、MF、KEGG 三合一图作法三合一的图需要以下格式： 只取Term和PValue的值，其中Term需要去除前面GO：xxxx的数值，留下后面的信号通路名字 同样的道理，分别搜集BP、CC、MF并累积到一个Sheet中，如图所示： subgroup与Enrichment Score需要自己手动创建，其中subgroup对应的就是你搜集下来的数据是属于BP还是CC亦或者是MF。需要作一个提前标识。而Enrichment Score列...</div></div></div></a><a class="pagination-related" href="/posts/20250831205237.html" title="RNA-seq基本介绍及实战——表达定量"><div class="cover" style="background:var(--default-bg-color)"></div><div class="info text-right"><div class="info-1"><div class="info-item-1">下一篇</div><div class="info-item-2">RNA-seq基本介绍及实战——表达定量</div></div><div class="info-2"><div class="info-item-1">实战环节上一期对所拥有的数据做了个序列比对，现在进入表达定量的环节 表达定量在进行表达定量的处理之前，需要对原始的比对文件进行处理，这里就有以下步骤： 使用picard &#x2F; samtools 将sam格式转换为bam格式 对bam文件进行排序 去除比对得分较低的序列 如果需要可以去除重复reads 在这里将会以三种方法进行表达定量的操作，分别是STAR+RSEM进行表达定量，另外一个就是使用Kallisto进行表达定量操作。最后一个就是使用featureCounts软件进行操作。 STAR+RSEM这个方法分为两个步骤 准定定量分析所需文件 利用STAR结果进行定量分析 在进行这个方法之前，需要对RSEM这个软件进行安装 RSEM这个软件的安装方法同样也很简单： 123456## 下载 RSEMwget -c https://github.com/deweylab/RSEM/archive/v1.3.1.tar.gzcd RSEM-1.3.1## 安装 RSEMmakemake install 接下来构建准备文件，在主目录下创建arab_RSEM文件夹，随后输...</div></div></div></a></nav><div class="relatedPosts"><div class="headline"><i class="fas fa-thumbs-up fa-fw"></i><span>相关推荐</span></div><div class="relatedPosts-list"><a class="pagination-related" href="/posts/20250821165237.html" title="RNA-seq基本介绍及实战——基本介绍"><div class="cover" style="background:var(--default-bg-color)"></div><div class="info text-center"><div class="info-1"><div class="info-item-1"><i class="far fa-calendar-alt fa-fw"></i> 2025-08-21</div><div class="info-item-2">RNA-seq基本介绍及实战——基本介绍</div></div><div class="info-2"><div class="info-item-1">RNA-seq的基本原理一、概念RNA-seq 是利用高通量测序技术对细胞或组织中 转录组（全部 RNA 分子集合） 进行测定的方法。它能揭示基因表达水平、转录本结构、可变剪接情况、新转录本等信息。相比于传统的 微阵列（microarray），RNA-seq 不依赖预先设计的探针，分辨率更高、动态范围更宽。 二、基本流程与原理 RNA 提取 从细胞或组织样本中提取总 RNA。 常常会去除 rRNA（占比高达 80-90%），保留 mRNA 或其他关注的 RNA 类型（如 miRNA、lncRNA）。 RNA → cDNA 由于测序平台主要针对 DNA，需要先把 RNA 逆转录为 cDNA。 通过 逆转录酶 合成一链或双链 cDNA。 文库构建 将 cDNA 打断成合适长度的片段（通常 200–500 bp）。 在片段两端连接 接头序列（adaptor），用于后续扩增和测序。 高通量测序 常见平台：Illumina（短读长，覆盖率高）、PacBio、Nanopore（长读长，适合全长转录本）。 测序得到大量的 reads（序列读段）。 数据分析 基因表达分析 （1...</div></div></div></a><a class="pagination-related" href="/posts/20250828165237.html" title="RNA-seq基本介绍及实战——质量控制"><div class="cover" style="background:var(--default-bg-color)"></div><div class="info text-center"><div class="info-1"><div class="info-item-1"><i class="far fa-calendar-alt fa-fw"></i> 2025-08-28</div><div class="info-item-2">RNA-seq基本介绍及实战——质量控制</div></div><div class="info-2"><div class="info-item-1">实战环节上一期对转录组学做了一个基本介绍，现在开始进入实战环节 数据预处理数据预处理在做之前，需要作准备工作，而准备工作一般是准备以下工作 准备工作 构建项目目录 参考序列下载 原始数据上传 构建项目目录进行转录组分析所使用的平台一般是linux系统，一个常见的工作目录结构如下： 参考序列下载参考序列一般来说我们需要两个文件 参考基因组（fasta格式） 注释信息 （gtf&#x2F;gff格式） 参考序列可以在ensemble数据库获得 里面包含了人类，小鼠等基因组的数据； 另外可访问JGI数据库 本次实例所用的数据库为TAIR数据库、对拟南芥基因库进行下载。 进入00ref目录，用wget命令进行下载 123wget https://plantgarden.jp/en/download/Arabidopsis_thaliana/t3702.G001/Araport11_GFF3_genes_transposons.201606.gtf.gzwget https://plantgarden.jp/en/download/Arabidopsis_thaliana/...</div></div></div></a><a class="pagination-related" href="/posts/20250830165237.html" title="RNA-seq基本介绍及实战——序列比对"><div class="cover" style="background:var(--default-bg-color)"></div><div class="info text-center"><div class="info-1"><div class="info-item-1"><i class="far fa-calendar-alt fa-fw"></i> 2025-08-30</div><div class="info-item-2">RNA-seq基本介绍及实战——序列比对</div></div><div class="info-2"><div class="info-item-1">实战环节上一期对所拥有的数据做了个初步数据的质量控制与过滤，现在开始进入序列比对环节 有参分析与无参分析 序列比对的流程如图所示 我们拿到测序的结果是一个个的片段，如果我们要获得这些片段的定量表达，就要知道这些片段，在参考基因组的哪个位置上。如果参考的物种有基因组的话，那我们就可以将这些片段比对到基因组上。这就是如图所示在左边的有参分析。 反之如果没有参考基因组的话，那就要进行转录组的拼接。直接利用测序读长之间的重叠关系，从头拼接、组装出完整的序列（Contigs&#x2F;Scaffolds）。这就是如图所示右边的无参分析。 特征 有参分析 无参分析 核心需求 已有参考基因组 无参考基因组 基本原理 将短序列映射到参考序列上 利用序列重叠关系从头拼接 计算效率 高，速度快，资源消耗少 低，速度慢，资源消耗巨大 技术难度 相对较低，流程标准化 高，需要大量调试和优化 结果形式 SAM&#x2F;BAM（比对位置信息） FASTA（组装出的序列） 主要优势 高效、准确、易于下游分析 能发现全新遗传信息，不依赖参考序列 主要局限 依赖参考基因组质量...</div></div></div></a><a class="pagination-related" href="/posts/20250831205237.html" title="RNA-seq基本介绍及实战——表达定量"><div class="cover" style="background:var(--default-bg-color)"></div><div class="info text-center"><div class="info-1"><div class="info-item-1"><i class="far fa-calendar-alt fa-fw"></i> 2025-08-31</div><div class="info-item-2">RNA-seq基本介绍及实战——表达定量</div></div><div class="info-2"><div class="info-item-1">实战环节上一期对所拥有的数据做了个序列比对，现在进入表达定量的环节 表达定量在进行表达定量的处理之前，需要对原始的比对文件进行处理，这里就有以下步骤： 使用picard &#x2F; samtools 将sam格式转换为bam格式 对bam文件进行排序 去除比对得分较低的序列 如果需要可以去除重复reads 在这里将会以三种方法进行表达定量的操作，分别是STAR+RSEM进行表达定量，另外一个就是使用Kallisto进行表达定量操作。最后一个就是使用featureCounts软件进行操作。 STAR+RSEM这个方法分为两个步骤 准定定量分析所需文件 利用STAR结果进行定量分析 在进行这个方法之前，需要对RSEM这个软件进行安装 RSEM这个软件的安装方法同样也很简单： 123456## 下载 RSEMwget -c https://github.com/deweylab/RSEM/archive/v1.3.1.tar.gzcd RSEM-1.3.1## 安装 RSEMmakemake install 接下来构建准备文件，在主目录下创建arab_RSEM文件夹，随后输...</div></div></div></a><a class="pagination-related" href="/posts/20250818162836.html" title="R语言基础语法学习记录"><div class="cover" style="background:var(--default-bg-color)"></div><div class="info text-center"><div class="info-1"><div class="info-item-1"><i class="far fa-calendar-alt fa-fw"></i> 2025-08-18</div><div class="info-item-2">R语言基础语法学习记录</div></div><div class="info-2"><div class="info-item-1">前言在前些年，笔者主要使用的是python语言进行数据的分析工作。在后面到研二期间进行生物数据的处理时，发现R语言在转录组分析或代谢组学分析的场景应用更加广泛，基于以上需求，遂开始对R语言的语法与数据结构进行学习并整理。 R包的安装常用模块的安装在R语言中是使用以下命令进行安装 1install.packages(&quot;xxx&quot;) R包的更新 1update.packages(&quot;xxx&quot;) 安装与应用的实例 12345if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE)) install.packages(&quot;BiocManager&quot;)BiocManager::install(&quot;DESeq2&quot;)library(DESeq2) 在这里是先通过安装BiocManager，再通过BiocManager安装DESeq2的模块，而后引用DESeq2这个包 R包的使用引用 1library(xxx) 展示使用文档 123help(pa...</div></div></div></a><a class="pagination-related" href="/posts/20250823203726.html" title="在浏览器中运行Python--基于Vercel中部署JupyterLite过程"><div class="cover" style="background:var(--default-bg-color)"></div><div class="info text-center"><div class="info-1"><div class="info-item-1"><i class="far fa-calendar-alt fa-fw"></i> 2025-08-23</div><div class="info-item-2">在浏览器中运行Python--基于Vercel中部署JupyterLite过程</div></div><div class="info-2"><div class="info-item-1">介绍JupyterLite 是一个完全在浏览器中运行的 JupyterLab 发行版，它基于 JupyterLab 组件和扩展从头构建而成。 JupyterLite 可以在不需要安装任何软件的情况下直接在浏览器中运行。 本文记录了如何基于Vercel去部署JupyterLite站点 官网：https://jupyterlite.readthedocs.io/en/stable/index.html 部署过程在github中拉取jupyterlite Demo首先在GitHub中将该项目fork下来，地址：https://github.com/jupyterlite/demo 而后创建部署脚本，取名为deploy.sh 内容如下： 12345678910111213141516#!/bin/bashecho y | yum install wgetwget -qO- https://micromamba.snakepit.net/api/micromamba/linux-64/latest | tar -xvj bin/micromamba# activate the envir...</div></div></div></a></div></div><hr class="custom-hr"><div id="post-comment"><div class="comment-head"><div class="comment-headline"><i class="fas fa-comments fa-fw"></i><span> 评论</span></div></div><div class="comment-wrap"><div><div id="twikoo-wrap"></div></div></div></div></div><div class="aside-content" id="aside-content"><div class="card-widget card-info text-center"><div class="avatar-img"><img src="/img/butterfly-icon.jpg" onerror='this.onerror=null,this.src="/img/friend_404.gif"' alt="avatar"></div><div class="author-info-name">Jinheng Hao</div><div class="author-info-description">生信/实验/技术交流</div><div class="site-data"><a href="/archives/"><div class="headline">文章</div><div class="length-num">23</div></a><a href="/tags/"><div class="headline">标签</div><div class="length-num">14</div></a><a href="/categories/"><div class="headline">分类</div><div class="length-num">7</div></a></div><a id="card-info-btn" target="_blank" rel="noopener external nofollow noreferrer" href="https://github.com/QDcvd"><i class="fab fa-github"></i><span>Follow Me</span></a></div><div class="card-widget card-announcement"><div class="item-headline"><i class="fas fa-bullhorn fa-shake"></i><span>公告</span></div><div class="announcement_content">如果你对内容有什么疑问，请积极评论</div></div><div class="sticky_layout"><div class="card-widget" id="card-toc"><div class="item-headline"><i class="fas fa-stream"></i><span>目录</span><span class="toc-percentage"></span></div><div class="toc-content"><ol class="toc"><li class="toc-item toc-level-2"><a class="toc-link" href="#%E5%AE%9E%E6%88%98%E7%8E%AF%E8%8A%82"><span class="toc-number">1.</span> <span class="toc-text">实战环节</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E5%B7%AE%E5%BC%82%E5%88%86%E6%9E%90"><span class="toc-number">2.</span> <span class="toc-text">差异分析</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E4%BD%BF%E7%94%A8omicverse%E8%BF%9B%E8%A1%8C%E5%B8%B8%E8%A7%84%E7%9A%84%E5%B7%AE%E5%BC%82%E5%9F%BA%E5%9B%A0%E8%A1%A8%E8%BE%BE%E5%88%86%E6%9E%90"><span class="toc-number">3.</span> <span class="toc-text">使用omicverse进行常规的差异基因表达分析</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E4%BD%BF%E7%94%A8-DEseq2-%E8%BF%9B%E8%A1%8C%E5%B7%AE%E5%BC%82%E8%A1%A8%E8%BE%BE%E5%88%86%E6%9E%90"><span class="toc-number">4.</span> <span class="toc-text">使用 DEseq2 进行差异表达分析</span></a></li></ol></div></div><div class="card-widget card-recent-post"><div class="item-headline"><i class="fas fa-history"></i><span>最新文章</span></div><div class="aside-list"><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/posts/20250930200124.html" title="SQL 实战入门：DDL数据库操作+三大数据类型全解析">SQL 实战入门：DDL数据库操作+三大数据类型全解析</a><time datetime="2025-09-30T12:01:24.000Z" title="发表于 2025-09-30 20:01:24">2025-09-30</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/posts/20250926164634.html" title="二代测序数据基础分析(一)">二代测序数据基础分析(一)</a><time datetime="2025-09-26T08:46:34.000Z" title="发表于 2025-09-26 16:46:34">2025-09-26</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/posts/20250924211902.html" title="python导入模块时遇到ModuleNotFoundError: No module named ’xxx‘">python导入模块时遇到ModuleNotFoundError: No module named ’xxx‘</a><time datetime="2025-09-24T13:19:02.000Z" title="发表于 2025-09-24 21:19:02">2025-09-24</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/posts/20250921164632.html" title="秀丽隐杆线虫油红O响应值测定方法">秀丽隐杆线虫油红O响应值测定方法</a><time datetime="2025-09-21T08:46:32.000Z" title="发表于 2025-09-21 16:46:32">2025-09-21</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/posts/20250916200356.html" title="基于David数据库与微生信平台进行Go-KEGG分析">基于David数据库与微生信平台进行Go-KEGG分析</a><time datetime="2025-09-16T12:03:56.000Z" title="发表于 2025-09-16 20:03:56">2025-09-16</time></div></div></div></div></div></div></main><footer id="footer"><div class="footer-other"><div class="footer-copyright"></div></div></footer></div><div id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="阅读模式"><i class="fas fa-book-open"></i></button><button id="darkmode" type="button" title="日间和夜间模式切换"><i class="fas fa-adjust"></i></button><button id="hide-aside-btn" type="button" title="单栏和双栏切换"><i class="fas fa-arrows-alt-h"></i></button></div><div id="rightside-config-show"><button id="rightside-config" type="button" title="设置"><i class="fas fa-cog fa-spin"></i></button><button class="close" id="mobile-toc-button" type="button" title="目录"><i class="fas fa-list-ul"></i></button><a id="to_comment" href="#post-comment" title="前往评论"><i class="fas fa-comments"></i></a><button id="go-up" type="button" title="回到顶部"><span class="scroll-percent"></span><i class="fas fa-arrow-up"></i></button></div></div><div><script src="/js/utils.js"></script><script src="/js/main.js"></script><script src="https://cdn.jsdelivr.net/npm/node-snackbar/dist/snackbar.min.js"></script><div class="js-pjax"><script>(()=>{let e="shuoshuo"===GLOBAL_CONFIG_SITE.pageType,n=null,i=()=>{let t=document.getElementById("twikoo-count");t&&twikoo.getCommentsCount({envId:"https://blogcomment0516.netlify.app/.netlify/functions/twikoo",region:"",urls:[window.location.pathname],includeReply:!1}).then(o=>{t.textContent=o[0].count}).catch(o=>{console.error(o)})},l=(o=document,t=location.pathname)=>{twikoo.init({el:o.querySelector("#twikoo-wrap"),envId:"https://blogcomment0516.netlify.app/.netlify/functions/twikoo",region:"",onCommentLoaded:()=>{btf.loadLightbox(document.querySelectorAll("#twikoo .tk-content img:not(.tk-owo-emotion)"))},...n,path:!e&&n&&n.path||t}),"post"===GLOBAL_CONFIG_SITE.pageType&&i(),e&&(window.shuoshuoComment.destroyTwikoo=()=>{o.children.length&&(o.innerHTML="",o.classList.add("no-comment"))})};var o=(o,t)=>{"object"==typeof twikoo?setTimeout(()=>l(o,t),0):btf.getScript("https://cdn.jsdelivr.net/npm/twikoo/dist/twikoo.all.min.js").then(()=>l(o,t))};e?window.shuoshuoComment={loadComment:o}:o()})()</script></div><canvas class="fireworks" mobile="false"></canvas><script src="https://cdn.jsdelivr.net/npm/butterfly-extsrc/dist/fireworks.min.js"></script><script defer id="ribbon" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc/dist/canvas-ribbon.min.js" size="150" alpha="0.6" zindex="-1" mobile="false" data-click="false"></script><script async data-pjax src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script></div></body></html>